{"code":422,"data":[{"message":"E-Mail is invalid."}],"level":"error","message":"An error occurred while creating a user: Invalid input.","stack":"Error: Invalid input.\n    at createUser (C:\\Note Taking App\\src\\controllers\\userController.js:21:21)\n    at Layer.handle [as handle_request] (C:\\Note Taking App\\node_modules\\express\\lib\\router\\layer.js:95:5)\n    at next (C:\\Note Taking App\\node_modules\\express\\lib\\router\\route.js:144:13)\n    at Route.dispatch (C:\\Note Taking App\\node_modules\\express\\lib\\router\\route.js:114:3)\n    at Layer.handle [as handle_request] (C:\\Note Taking App\\node_modules\\express\\lib\\router\\layer.js:95:5)\n    at C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:284:15\n    at Function.process_params (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:346:12)\n    at next (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:280:10)\n    at Function.handle (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:175:3)\n    at router (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:47:12)\n    at Layer.handle [as handle_request] (C:\\Note Taking App\\node_modules\\express\\lib\\router\\layer.js:95:5)\n    at trim_prefix (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:328:13)\n    at C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:286:9\n    at Function.process_params (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:346:12)\n    at next (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:280:10)\n    at C:\\Note Taking App\\node_modules\\body-parser\\lib\\read.js:137:5\n    at AsyncResource.runInAsyncScope (node:async_hooks:201:9)\n    at invokeCallback (C:\\Note Taking App\\node_modules\\raw-body\\index.js:238:16)\n    at done (C:\\Note Taking App\\node_modules\\raw-body\\index.js:227:7)\n    at IncomingMessage.onEnd (C:\\Note Taking App\\node_modules\\raw-body\\index.js:287:7)\n    at IncomingMessage.emit (node:events:538:35)\n    at endReadableNT (node:internal/streams/readable:1345:12)\n    at processTicksAndRejections (node:internal/process/task_queues:83:21)","timestamp":"2023-05-22T09:25:29.064Z"}
{"code":422,"data":[{"message":"E-Mail is invalid."}],"level":"error","message":"An error occurred while updating a user: Invalid input.","stack":"Error: Invalid input.\n    at updateUser (C:\\Note Taking App\\src\\controllers\\userController.js:66:21)\n    at Layer.handle [as handle_request] (C:\\Note Taking App\\node_modules\\express\\lib\\router\\layer.js:95:5)\n    at next (C:\\Note Taking App\\node_modules\\express\\lib\\router\\route.js:144:13)\n    at Route.dispatch (C:\\Note Taking App\\node_modules\\express\\lib\\router\\route.js:114:3)\n    at Layer.handle [as handle_request] (C:\\Note Taking App\\node_modules\\express\\lib\\router\\layer.js:95:5)\n    at C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:284:15\n    at param (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:365:14)\n    at param (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:376:14)\n    at Function.process_params (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:421:3)\n    at next (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:280:10)\n    at Function.handle (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:175:3)\n    at router (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:47:12)\n    at Layer.handle [as handle_request] (C:\\Note Taking App\\node_modules\\express\\lib\\router\\layer.js:95:5)\n    at trim_prefix (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:328:13)\n    at C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:286:9\n    at Function.process_params (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:346:12)\n    at next (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:280:10)\n    at C:\\Note Taking App\\node_modules\\body-parser\\lib\\read.js:137:5\n    at AsyncResource.runInAsyncScope (node:async_hooks:201:9)\n    at invokeCallback (C:\\Note Taking App\\node_modules\\raw-body\\index.js:238:16)\n    at done (C:\\Note Taking App\\node_modules\\raw-body\\index.js:227:7)\n    at IncomingMessage.onEnd (C:\\Note Taking App\\node_modules\\raw-body\\index.js:287:7)\n    at IncomingMessage.emit (node:events:538:35)\n    at endReadableNT (node:internal/streams/readable:1345:12)\n    at processTicksAndRejections (node:internal/process/task_queues:83:21)","timestamp":"2023-05-22T09:25:29.310Z"}
{"code":422,"data":[{"message":"E-Mail is invalid."}],"level":"error","message":"An error occurred while creating a user: Invalid input.","stack":"Error: Invalid input.\n    at createUser (C:\\Note Taking App\\src\\controllers\\userController.js:22:21)\n    at Layer.handle [as handle_request] (C:\\Note Taking App\\node_modules\\express\\lib\\router\\layer.js:95:5)\n    at next (C:\\Note Taking App\\node_modules\\express\\lib\\router\\route.js:144:13)\n    at Route.dispatch (C:\\Note Taking App\\node_modules\\express\\lib\\router\\route.js:114:3)\n    at Layer.handle [as handle_request] (C:\\Note Taking App\\node_modules\\express\\lib\\router\\layer.js:95:5)\n    at C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:284:15\n    at Function.process_params (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:346:12)\n    at next (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:280:10)\n    at Function.handle (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:175:3)\n    at router (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:47:12)\n    at Layer.handle [as handle_request] (C:\\Note Taking App\\node_modules\\express\\lib\\router\\layer.js:95:5)\n    at trim_prefix (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:328:13)\n    at C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:286:9\n    at Function.process_params (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:346:12)\n    at next (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:280:10)\n    at C:\\Note Taking App\\node_modules\\body-parser\\lib\\read.js:137:5\n    at AsyncResource.runInAsyncScope (node:async_hooks:201:9)\n    at invokeCallback (C:\\Note Taking App\\node_modules\\raw-body\\index.js:238:16)\n    at done (C:\\Note Taking App\\node_modules\\raw-body\\index.js:227:7)\n    at IncomingMessage.onEnd (C:\\Note Taking App\\node_modules\\raw-body\\index.js:287:7)\n    at IncomingMessage.emit (node:events:538:35)\n    at endReadableNT (node:internal/streams/readable:1345:12)\n    at processTicksAndRejections (node:internal/process/task_queues:83:21)","timestamp":"2023-05-22T09:37:22.957Z"}
{"code":422,"data":[{"message":"E-Mail is invalid."}],"level":"error","message":"An error occurred while updating a user: Invalid input.","stack":"Error: Invalid input.\n    at updateUser (C:\\Note Taking App\\src\\controllers\\userController.js:67:21)\n    at Layer.handle [as handle_request] (C:\\Note Taking App\\node_modules\\express\\lib\\router\\layer.js:95:5)\n    at next (C:\\Note Taking App\\node_modules\\express\\lib\\router\\route.js:144:13)\n    at Route.dispatch (C:\\Note Taking App\\node_modules\\express\\lib\\router\\route.js:114:3)\n    at Layer.handle [as handle_request] (C:\\Note Taking App\\node_modules\\express\\lib\\router\\layer.js:95:5)\n    at C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:284:15\n    at param (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:365:14)\n    at param (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:376:14)\n    at Function.process_params (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:421:3)\n    at next (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:280:10)\n    at Function.handle (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:175:3)\n    at router (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:47:12)\n    at Layer.handle [as handle_request] (C:\\Note Taking App\\node_modules\\express\\lib\\router\\layer.js:95:5)\n    at trim_prefix (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:328:13)\n    at C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:286:9\n    at Function.process_params (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:346:12)\n    at next (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:280:10)\n    at C:\\Note Taking App\\node_modules\\body-parser\\lib\\read.js:137:5\n    at AsyncResource.runInAsyncScope (node:async_hooks:201:9)\n    at invokeCallback (C:\\Note Taking App\\node_modules\\raw-body\\index.js:238:16)\n    at done (C:\\Note Taking App\\node_modules\\raw-body\\index.js:227:7)\n    at IncomingMessage.onEnd (C:\\Note Taking App\\node_modules\\raw-body\\index.js:287:7)\n    at IncomingMessage.emit (node:events:538:35)\n    at endReadableNT (node:internal/streams/readable:1345:12)\n    at processTicksAndRejections (node:internal/process/task_queues:83:21)","timestamp":"2023-05-22T09:37:29.040Z"}
{"level":"error","message":"An error occurred while creating a user: Operation `users.insertOne()` buffering timed out after 10000ms","stack":"MongooseError: Operation `users.insertOne()` buffering timed out after 10000ms\n    at Timeout.<anonymous> (C:\\Note Taking App\\node_modules\\mongoose\\lib\\drivers\\node-mongodb-native\\collection.js:185:23)\n    at listOnTimeout (node:internal/timers:559:17)\n    at processTimers (node:internal/timers:502:7)","timestamp":"2023-05-22T09:37:31.017Z"}
{"level":"error","message":"An error occurred while fetching user by ID: Operation `users.findOne()` buffering timed out after 10000ms","stack":"MongooseError: Operation `users.findOne()` buffering timed out after 10000ms\n    at Timeout.<anonymous> (C:\\Note Taking App\\node_modules\\mongoose\\lib\\drivers\\node-mongodb-native\\collection.js:185:23)\n    at listOnTimeout (node:internal/timers:559:17)\n    at processTimers (node:internal/timers:502:7)","timestamp":"2023-05-22T09:37:33.039Z"}
{"level":"error","message":"An error occurred while fetching user by ID: Operation `users.findOne()` buffering timed out after 10000ms","stack":"MongooseError: Operation `users.findOne()` buffering timed out after 10000ms\n    at Timeout.<anonymous> (C:\\Note Taking App\\node_modules\\mongoose\\lib\\drivers\\node-mongodb-native\\collection.js:185:23)\n    at listOnTimeout (node:internal/timers:559:17)\n    at processTimers (node:internal/timers:502:7)","timestamp":"2023-05-22T09:37:35.020Z"}
{"code":422,"data":[{"message":"E-Mail is invalid."}],"level":"error","message":"An error occurred while creating a user: Invalid input.","stack":"Error: Invalid input.\n    at createUser (C:\\Note Taking App\\src\\controllers\\userController.js:22:21)\n    at Layer.handle [as handle_request] (C:\\Note Taking App\\node_modules\\express\\lib\\router\\layer.js:95:5)\n    at next (C:\\Note Taking App\\node_modules\\express\\lib\\router\\route.js:144:13)\n    at Route.dispatch (C:\\Note Taking App\\node_modules\\express\\lib\\router\\route.js:114:3)\n    at Layer.handle [as handle_request] (C:\\Note Taking App\\node_modules\\express\\lib\\router\\layer.js:95:5)\n    at C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:284:15\n    at Function.process_params (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:346:12)\n    at next (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:280:10)\n    at Function.handle (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:175:3)\n    at router (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:47:12)\n    at Layer.handle [as handle_request] (C:\\Note Taking App\\node_modules\\express\\lib\\router\\layer.js:95:5)\n    at trim_prefix (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:328:13)\n    at C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:286:9\n    at Function.process_params (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:346:12)\n    at next (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:280:10)\n    at C:\\Note Taking App\\node_modules\\body-parser\\lib\\read.js:137:5\n    at AsyncResource.runInAsyncScope (node:async_hooks:201:9)\n    at invokeCallback (C:\\Note Taking App\\node_modules\\raw-body\\index.js:238:16)\n    at done (C:\\Note Taking App\\node_modules\\raw-body\\index.js:227:7)\n    at IncomingMessage.onEnd (C:\\Note Taking App\\node_modules\\raw-body\\index.js:287:7)\n    at IncomingMessage.emit (node:events:538:35)\n    at endReadableNT (node:internal/streams/readable:1345:12)\n    at processTicksAndRejections (node:internal/process/task_queues:83:21)","timestamp":"2023-05-22T09:38:35.703Z"}
{"code":422,"data":[{"message":"E-Mail is invalid."}],"level":"error","message":"An error occurred while updating a user: Invalid input.","stack":"Error: Invalid input.\n    at updateUser (C:\\Note Taking App\\src\\controllers\\userController.js:67:21)\n    at Layer.handle [as handle_request] (C:\\Note Taking App\\node_modules\\express\\lib\\router\\layer.js:95:5)\n    at next (C:\\Note Taking App\\node_modules\\express\\lib\\router\\route.js:144:13)\n    at Route.dispatch (C:\\Note Taking App\\node_modules\\express\\lib\\router\\route.js:114:3)\n    at Layer.handle [as handle_request] (C:\\Note Taking App\\node_modules\\express\\lib\\router\\layer.js:95:5)\n    at C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:284:15\n    at param (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:365:14)\n    at param (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:376:14)\n    at Function.process_params (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:421:3)\n    at next (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:280:10)\n    at Function.handle (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:175:3)\n    at router (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:47:12)\n    at Layer.handle [as handle_request] (C:\\Note Taking App\\node_modules\\express\\lib\\router\\layer.js:95:5)\n    at trim_prefix (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:328:13)\n    at C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:286:9\n    at Function.process_params (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:346:12)\n    at next (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:280:10)\n    at C:\\Note Taking App\\node_modules\\body-parser\\lib\\read.js:137:5\n    at AsyncResource.runInAsyncScope (node:async_hooks:201:9)\n    at invokeCallback (C:\\Note Taking App\\node_modules\\raw-body\\index.js:238:16)\n    at done (C:\\Note Taking App\\node_modules\\raw-body\\index.js:227:7)\n    at IncomingMessage.onEnd (C:\\Note Taking App\\node_modules\\raw-body\\index.js:287:7)\n    at IncomingMessage.emit (node:events:538:35)\n    at endReadableNT (node:internal/streams/readable:1345:12)\n    at processTicksAndRejections (node:internal/process/task_queues:83:21)","timestamp":"2023-05-22T09:38:39.780Z"}
{"code":422,"data":[{"message":"E-Mail is invalid."}],"level":"error","message":"An error occurred while creating a user: Invalid input.","stack":"Error: Invalid input.\n    at createUser (C:\\Note Taking App\\src\\controllers\\userController.js:22:21)\n    at Layer.handle [as handle_request] (C:\\Note Taking App\\node_modules\\express\\lib\\router\\layer.js:95:5)\n    at next (C:\\Note Taking App\\node_modules\\express\\lib\\router\\route.js:144:13)\n    at Route.dispatch (C:\\Note Taking App\\node_modules\\express\\lib\\router\\route.js:114:3)\n    at Layer.handle [as handle_request] (C:\\Note Taking App\\node_modules\\express\\lib\\router\\layer.js:95:5)\n    at C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:284:15\n    at Function.process_params (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:346:12)\n    at next (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:280:10)\n    at Function.handle (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:175:3)\n    at router (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:47:12)\n    at Layer.handle [as handle_request] (C:\\Note Taking App\\node_modules\\express\\lib\\router\\layer.js:95:5)\n    at trim_prefix (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:328:13)\n    at C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:286:9\n    at Function.process_params (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:346:12)\n    at next (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:280:10)\n    at C:\\Note Taking App\\node_modules\\body-parser\\lib\\read.js:137:5\n    at AsyncResource.runInAsyncScope (node:async_hooks:201:9)\n    at invokeCallback (C:\\Note Taking App\\node_modules\\raw-body\\index.js:238:16)\n    at done (C:\\Note Taking App\\node_modules\\raw-body\\index.js:227:7)\n    at IncomingMessage.onEnd (C:\\Note Taking App\\node_modules\\raw-body\\index.js:287:7)\n    at IncomingMessage.emit (node:events:538:35)\n    at endReadableNT (node:internal/streams/readable:1345:12)\n    at processTicksAndRejections (node:internal/process/task_queues:83:21)","timestamp":"2023-05-22T10:03:55.213Z"}
{"code":422,"data":[{"message":"E-Mail is invalid."}],"level":"error","message":"An error occurred while updating a user: Invalid input.","stack":"Error: Invalid input.\n    at updateUser (C:\\Note Taking App\\src\\controllers\\userController.js:67:21)\n    at Layer.handle [as handle_request] (C:\\Note Taking App\\node_modules\\express\\lib\\router\\layer.js:95:5)\n    at next (C:\\Note Taking App\\node_modules\\express\\lib\\router\\route.js:144:13)\n    at Route.dispatch (C:\\Note Taking App\\node_modules\\express\\lib\\router\\route.js:114:3)\n    at Layer.handle [as handle_request] (C:\\Note Taking App\\node_modules\\express\\lib\\router\\layer.js:95:5)\n    at C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:284:15\n    at param (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:365:14)\n    at param (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:376:14)\n    at Function.process_params (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:421:3)\n    at next (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:280:10)\n    at Function.handle (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:175:3)\n    at router (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:47:12)\n    at Layer.handle [as handle_request] (C:\\Note Taking App\\node_modules\\express\\lib\\router\\layer.js:95:5)\n    at trim_prefix (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:328:13)\n    at C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:286:9\n    at Function.process_params (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:346:12)\n    at next (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:280:10)\n    at C:\\Note Taking App\\node_modules\\body-parser\\lib\\read.js:137:5\n    at AsyncResource.runInAsyncScope (node:async_hooks:201:9)\n    at invokeCallback (C:\\Note Taking App\\node_modules\\raw-body\\index.js:238:16)\n    at done (C:\\Note Taking App\\node_modules\\raw-body\\index.js:227:7)\n    at IncomingMessage.onEnd (C:\\Note Taking App\\node_modules\\raw-body\\index.js:287:7)\n    at IncomingMessage.emit (node:events:538:35)\n    at endReadableNT (node:internal/streams/readable:1345:12)\n    at processTicksAndRejections (node:internal/process/task_queues:83:21)","timestamp":"2023-05-22T10:04:01.277Z"}
{"code":422,"data":[{"message":"E-Mail is invalid."}],"level":"error","message":"An error occurred while creating a user: Invalid input.","stack":"Error: Invalid input.\n    at createUser (C:\\Note Taking App\\src\\controllers\\userController.js:22:21)\n    at Layer.handle [as handle_request] (C:\\Note Taking App\\node_modules\\express\\lib\\router\\layer.js:95:5)\n    at next (C:\\Note Taking App\\node_modules\\express\\lib\\router\\route.js:144:13)\n    at Route.dispatch (C:\\Note Taking App\\node_modules\\express\\lib\\router\\route.js:114:3)\n    at Layer.handle [as handle_request] (C:\\Note Taking App\\node_modules\\express\\lib\\router\\layer.js:95:5)\n    at C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:284:15\n    at Function.process_params (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:346:12)\n    at next (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:280:10)\n    at Function.handle (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:175:3)\n    at router (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:47:12)\n    at Layer.handle [as handle_request] (C:\\Note Taking App\\node_modules\\express\\lib\\router\\layer.js:95:5)\n    at trim_prefix (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:328:13)\n    at C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:286:9\n    at Function.process_params (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:346:12)\n    at next (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:280:10)\n    at C:\\Note Taking App\\node_modules\\body-parser\\lib\\read.js:137:5\n    at AsyncResource.runInAsyncScope (node:async_hooks:201:9)\n    at invokeCallback (C:\\Note Taking App\\node_modules\\raw-body\\index.js:238:16)\n    at done (C:\\Note Taking App\\node_modules\\raw-body\\index.js:227:7)\n    at IncomingMessage.onEnd (C:\\Note Taking App\\node_modules\\raw-body\\index.js:287:7)\n    at IncomingMessage.emit (node:events:538:35)\n    at endReadableNT (node:internal/streams/readable:1345:12)\n    at processTicksAndRejections (node:internal/process/task_queues:83:21)","timestamp":"2023-05-22T10:15:35.802Z"}
{"code":422,"data":[{"message":"E-Mail is invalid."}],"level":"error","message":"An error occurred while updating a user: Invalid input.","stack":"Error: Invalid input.\n    at updateUser (C:\\Note Taking App\\src\\controllers\\userController.js:67:21)\n    at Layer.handle [as handle_request] (C:\\Note Taking App\\node_modules\\express\\lib\\router\\layer.js:95:5)\n    at next (C:\\Note Taking App\\node_modules\\express\\lib\\router\\route.js:144:13)\n    at Route.dispatch (C:\\Note Taking App\\node_modules\\express\\lib\\router\\route.js:114:3)\n    at Layer.handle [as handle_request] (C:\\Note Taking App\\node_modules\\express\\lib\\router\\layer.js:95:5)\n    at C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:284:15\n    at param (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:365:14)\n    at param (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:376:14)\n    at Function.process_params (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:421:3)\n    at next (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:280:10)\n    at Function.handle (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:175:3)\n    at router (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:47:12)\n    at Layer.handle [as handle_request] (C:\\Note Taking App\\node_modules\\express\\lib\\router\\layer.js:95:5)\n    at trim_prefix (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:328:13)\n    at C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:286:9\n    at Function.process_params (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:346:12)\n    at next (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:280:10)\n    at C:\\Note Taking App\\node_modules\\body-parser\\lib\\read.js:137:5\n    at AsyncResource.runInAsyncScope (node:async_hooks:201:9)\n    at invokeCallback (C:\\Note Taking App\\node_modules\\raw-body\\index.js:238:16)\n    at done (C:\\Note Taking App\\node_modules\\raw-body\\index.js:227:7)\n    at IncomingMessage.onEnd (C:\\Note Taking App\\node_modules\\raw-body\\index.js:287:7)\n    at IncomingMessage.emit (node:events:538:35)\n    at endReadableNT (node:internal/streams/readable:1345:12)\n    at processTicksAndRejections (node:internal/process/task_queues:83:21)","timestamp":"2023-05-22T10:15:36.035Z"}
{"code":422,"data":[{"message":"E-Mail is invalid."}],"level":"error","message":"An error occurred while creating a user: Invalid input.","stack":"Error: Invalid input.\n    at createUser (C:\\Note Taking App\\src\\controllers\\userController.js:22:21)\n    at Layer.handle [as handle_request] (C:\\Note Taking App\\node_modules\\express\\lib\\router\\layer.js:95:5)\n    at next (C:\\Note Taking App\\node_modules\\express\\lib\\router\\route.js:144:13)\n    at Route.dispatch (C:\\Note Taking App\\node_modules\\express\\lib\\router\\route.js:114:3)\n    at Layer.handle [as handle_request] (C:\\Note Taking App\\node_modules\\express\\lib\\router\\layer.js:95:5)\n    at C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:284:15\n    at Function.process_params (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:346:12)\n    at next (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:280:10)\n    at Function.handle (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:175:3)\n    at router (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:47:12)\n    at Layer.handle [as handle_request] (C:\\Note Taking App\\node_modules\\express\\lib\\router\\layer.js:95:5)\n    at trim_prefix (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:328:13)\n    at C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:286:9\n    at Function.process_params (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:346:12)\n    at next (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:280:10)\n    at C:\\Note Taking App\\node_modules\\body-parser\\lib\\read.js:137:5\n    at AsyncResource.runInAsyncScope (node:async_hooks:201:9)\n    at invokeCallback (C:\\Note Taking App\\node_modules\\raw-body\\index.js:238:16)\n    at done (C:\\Note Taking App\\node_modules\\raw-body\\index.js:227:7)\n    at IncomingMessage.onEnd (C:\\Note Taking App\\node_modules\\raw-body\\index.js:287:7)\n    at IncomingMessage.emit (node:events:538:35)\n    at endReadableNT (node:internal/streams/readable:1345:12)\n    at processTicksAndRejections (node:internal/process/task_queues:83:21)","timestamp":"2023-05-22T10:18:38.341Z"}
{"code":422,"data":[{"message":"E-Mail is invalid."}],"level":"error","message":"An error occurred while updating a user: Invalid input.","stack":"Error: Invalid input.\n    at updateUser (C:\\Note Taking App\\src\\controllers\\userController.js:67:21)\n    at Layer.handle [as handle_request] (C:\\Note Taking App\\node_modules\\express\\lib\\router\\layer.js:95:5)\n    at next (C:\\Note Taking App\\node_modules\\express\\lib\\router\\route.js:144:13)\n    at Route.dispatch (C:\\Note Taking App\\node_modules\\express\\lib\\router\\route.js:114:3)\n    at Layer.handle [as handle_request] (C:\\Note Taking App\\node_modules\\express\\lib\\router\\layer.js:95:5)\n    at C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:284:15\n    at param (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:365:14)\n    at param (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:376:14)\n    at Function.process_params (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:421:3)\n    at next (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:280:10)\n    at Function.handle (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:175:3)\n    at router (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:47:12)\n    at Layer.handle [as handle_request] (C:\\Note Taking App\\node_modules\\express\\lib\\router\\layer.js:95:5)\n    at trim_prefix (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:328:13)\n    at C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:286:9\n    at Function.process_params (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:346:12)\n    at next (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:280:10)\n    at C:\\Note Taking App\\node_modules\\body-parser\\lib\\read.js:137:5\n    at AsyncResource.runInAsyncScope (node:async_hooks:201:9)\n    at invokeCallback (C:\\Note Taking App\\node_modules\\raw-body\\index.js:238:16)\n    at done (C:\\Note Taking App\\node_modules\\raw-body\\index.js:227:7)\n    at IncomingMessage.onEnd (C:\\Note Taking App\\node_modules\\raw-body\\index.js:287:7)\n    at IncomingMessage.emit (node:events:538:35)\n    at endReadableNT (node:internal/streams/readable:1345:12)\n    at processTicksAndRejections (node:internal/process/task_queues:83:21)","timestamp":"2023-05-22T10:18:38.561Z"}
{"kind":"ObjectId","level":"error","message":"An error occurred while fetching user by ID: Cast to ObjectId failed for value \"nonexistentuserid\" (type string) at path \"_id\" for model \"User\"","path":"_id","reason":{},"stack":"CastError: Cast to ObjectId failed for value \"nonexistentuserid\" (type string) at path \"_id\" for model \"User\"\n    at ObjectId.cast (C:\\Note Taking App\\node_modules\\mongoose\\lib\\schema\\objectid.js:248:11)\n    at ObjectId.SchemaType.applySetters (C:\\Note Taking App\\node_modules\\mongoose\\lib\\schematype.js:1202:12)\n    at ObjectId.SchemaType.castForQuery (C:\\Note Taking App\\node_modules\\mongoose\\lib\\schematype.js:1614:15)\n    at cast (C:\\Note Taking App\\node_modules\\mongoose\\lib\\cast.js:348:32)\n    at model.Query.Query.cast (C:\\Note Taking App\\node_modules\\mongoose\\lib\\query.js:4819:12)\n    at model.Query.Query._castConditions (C:\\Note Taking App\\node_modules\\mongoose\\lib\\query.js:2193:10)\n    at model.Query._findOne (C:\\Note Taking App\\node_modules\\mongoose\\lib\\query.js:2477:8)\n    at model.Query.exec (C:\\Note Taking App\\node_modules\\mongoose\\lib\\query.js:4349:28)\n    at processTicksAndRejections (node:internal/process/task_queues:96:5)\n    at async Object.getUserById (C:\\Note Taking App\\src\\services\\userServices.js:10:16)\n    at async getUserById (C:\\Note Taking App\\src\\controllers\\userController.js:38:18)","stringValue":"\"nonexistentuserid\"","timestamp":"2023-05-22T10:47:18.840Z","value":"nonexistentuserid","valueType":"string"}
{"level":"error","message":"An error occurred while creating a user: Expected a string but received a undefined","stack":"TypeError: Expected a string but received a undefined\n    at assertString (C:\\Note Taking App\\node_modules\\validator\\lib\\util\\assertString.js:17:11)\n    at Object.isEmail (C:\\Note Taking App\\node_modules\\validator\\lib\\isEmail.js:79:29)\n    at createUser (C:\\Note Taking App\\src\\controllers\\userController.js:10:20)\n    at Layer.handle [as handle_request] (C:\\Note Taking App\\node_modules\\express\\lib\\router\\layer.js:95:5)\n    at next (C:\\Note Taking App\\node_modules\\express\\lib\\router\\route.js:144:13)\n    at Route.dispatch (C:\\Note Taking App\\node_modules\\express\\lib\\router\\route.js:114:3)\n    at Layer.handle [as handle_request] (C:\\Note Taking App\\node_modules\\express\\lib\\router\\layer.js:95:5)\n    at C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:284:15\n    at Function.process_params (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:346:12)\n    at next (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:280:10)\n    at Function.handle (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:175:3)\n    at router (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:47:12)\n    at Layer.handle [as handle_request] (C:\\Note Taking App\\node_modules\\express\\lib\\router\\layer.js:95:5)\n    at trim_prefix (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:328:13)\n    at C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:286:9\n    at Function.process_params (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:346:12)\n    at next (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:280:10)\n    at C:\\Note Taking App\\node_modules\\body-parser\\lib\\read.js:137:5\n    at AsyncResource.runInAsyncScope (node:async_hooks:201:9)\n    at invokeCallback (C:\\Note Taking App\\node_modules\\raw-body\\index.js:238:16)\n    at done (C:\\Note Taking App\\node_modules\\raw-body\\index.js:227:7)\n    at IncomingMessage.onEnd (C:\\Note Taking App\\node_modules\\raw-body\\index.js:287:7)\n    at IncomingMessage.emit (node:events:538:35)\n    at endReadableNT (node:internal/streams/readable:1345:12)\n    at processTicksAndRejections (node:internal/process/task_queues:83:21)","timestamp":"2023-05-22T10:47:19.462Z"}
{"level":"error","message":"An error occurred while updating a user: Expected a string but received a undefined","stack":"TypeError: Expected a string but received a undefined\n    at assertString (C:\\Note Taking App\\node_modules\\validator\\lib\\util\\assertString.js:17:11)\n    at Object.isEmail (C:\\Note Taking App\\node_modules\\validator\\lib\\isEmail.js:79:29)\n    at updateUser (C:\\Note Taking App\\src\\controllers\\userController.js:56:20)\n    at Layer.handle [as handle_request] (C:\\Note Taking App\\node_modules\\express\\lib\\router\\layer.js:95:5)\n    at next (C:\\Note Taking App\\node_modules\\express\\lib\\router\\route.js:144:13)\n    at Route.dispatch (C:\\Note Taking App\\node_modules\\express\\lib\\router\\route.js:114:3)\n    at Layer.handle [as handle_request] (C:\\Note Taking App\\node_modules\\express\\lib\\router\\layer.js:95:5)\n    at C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:284:15\n    at param (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:365:14)\n    at param (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:376:14)\n    at Function.process_params (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:421:3)\n    at next (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:280:10)\n    at Function.handle (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:175:3)\n    at router (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:47:12)\n    at Layer.handle [as handle_request] (C:\\Note Taking App\\node_modules\\express\\lib\\router\\layer.js:95:5)\n    at trim_prefix (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:328:13)\n    at C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:286:9\n    at Function.process_params (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:346:12)\n    at next (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:280:10)\n    at C:\\Note Taking App\\node_modules\\body-parser\\lib\\read.js:137:5\n    at AsyncResource.runInAsyncScope (node:async_hooks:201:9)\n    at invokeCallback (C:\\Note Taking App\\node_modules\\raw-body\\index.js:238:16)\n    at done (C:\\Note Taking App\\node_modules\\raw-body\\index.js:227:7)\n    at IncomingMessage.onEnd (C:\\Note Taking App\\node_modules\\raw-body\\index.js:287:7)\n    at IncomingMessage.emit (node:events:538:35)\n    at endReadableNT (node:internal/streams/readable:1345:12)\n    at processTicksAndRejections (node:internal/process/task_queues:83:21)","timestamp":"2023-05-22T10:47:19.501Z"}
{"kind":"ObjectId","level":"error","message":"An error occurred while deleting a user: Cast to ObjectId failed for value \"nonexistentuserid\" (type string) at path \"_id\" for model \"User\"","path":"_id","reason":{},"stack":"CastError: Cast to ObjectId failed for value \"nonexistentuserid\" (type string) at path \"_id\" for model \"User\"\n    at ObjectId.cast (C:\\Note Taking App\\node_modules\\mongoose\\lib\\schema\\objectid.js:248:11)\n    at ObjectId.SchemaType.applySetters (C:\\Note Taking App\\node_modules\\mongoose\\lib\\schematype.js:1202:12)\n    at ObjectId.SchemaType.castForQuery (C:\\Note Taking App\\node_modules\\mongoose\\lib\\schematype.js:1614:15)\n    at cast (C:\\Note Taking App\\node_modules\\mongoose\\lib\\cast.js:348:32)\n    at model.Query.Query.cast (C:\\Note Taking App\\node_modules\\mongoose\\lib\\query.js:4819:12)\n    at model.Query.Query._castConditions (C:\\Note Taking App\\node_modules\\mongoose\\lib\\query.js:2193:10)\n    at model.Query._findOneAndDelete (C:\\Note Taking App\\node_modules\\mongoose\\lib\\query.js:3422:8)\n    at model.Query._findOneAndRemove (C:\\Note Taking App\\node_modules\\mongoose\\lib\\query.js:3650:15)\n    at model.Query.exec (C:\\Note Taking App\\node_modules\\mongoose\\lib\\query.js:4349:28)\n    at processTicksAndRejections (node:internal/process/task_queues:96:5)\n    at async Object.deleteUser (C:\\Note Taking App\\src\\services\\userServices.js:21:3)\n    at async deleteUser (C:\\Note Taking App\\src\\controllers\\userController.js:93:5)","stringValue":"\"nonexistentuserid\"","timestamp":"2023-05-22T10:47:19.539Z","value":"nonexistentuserid","valueType":"string"}
{"code":422,"data":[{"message":"E-Mail is invalid."}],"level":"error","message":"An error occurred while creating a user: Invalid input.","stack":"Error: Invalid input.\n    at createUser (C:\\Note Taking App\\src\\controllers\\userController.js:22:21)\n    at Layer.handle [as handle_request] (C:\\Note Taking App\\node_modules\\express\\lib\\router\\layer.js:95:5)\n    at next (C:\\Note Taking App\\node_modules\\express\\lib\\router\\route.js:144:13)\n    at Route.dispatch (C:\\Note Taking App\\node_modules\\express\\lib\\router\\route.js:114:3)\n    at Layer.handle [as handle_request] (C:\\Note Taking App\\node_modules\\express\\lib\\router\\layer.js:95:5)\n    at C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:284:15\n    at Function.process_params (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:346:12)\n    at next (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:280:10)\n    at Function.handle (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:175:3)\n    at router (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:47:12)\n    at Layer.handle [as handle_request] (C:\\Note Taking App\\node_modules\\express\\lib\\router\\layer.js:95:5)\n    at trim_prefix (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:328:13)\n    at C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:286:9\n    at Function.process_params (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:346:12)\n    at next (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:280:10)\n    at C:\\Note Taking App\\node_modules\\body-parser\\lib\\read.js:137:5\n    at AsyncResource.runInAsyncScope (node:async_hooks:201:9)\n    at invokeCallback (C:\\Note Taking App\\node_modules\\raw-body\\index.js:238:16)\n    at done (C:\\Note Taking App\\node_modules\\raw-body\\index.js:227:7)\n    at IncomingMessage.onEnd (C:\\Note Taking App\\node_modules\\raw-body\\index.js:287:7)\n    at IncomingMessage.emit (node:events:538:35)\n    at endReadableNT (node:internal/streams/readable:1345:12)\n    at processTicksAndRejections (node:internal/process/task_queues:83:21)","timestamp":"2023-05-22T10:48:38.798Z"}
{"code":422,"data":[{"message":"E-Mail is invalid."}],"level":"error","message":"An error occurred while updating a user: Invalid input.","stack":"Error: Invalid input.\n    at updateUser (C:\\Note Taking App\\src\\controllers\\userController.js:67:21)\n    at Layer.handle [as handle_request] (C:\\Note Taking App\\node_modules\\express\\lib\\router\\layer.js:95:5)\n    at next (C:\\Note Taking App\\node_modules\\express\\lib\\router\\route.js:144:13)\n    at Route.dispatch (C:\\Note Taking App\\node_modules\\express\\lib\\router\\route.js:114:3)\n    at Layer.handle [as handle_request] (C:\\Note Taking App\\node_modules\\express\\lib\\router\\layer.js:95:5)\n    at C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:284:15\n    at param (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:365:14)\n    at param (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:376:14)\n    at Function.process_params (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:421:3)\n    at next (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:280:10)\n    at Function.handle (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:175:3)\n    at router (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:47:12)\n    at Layer.handle [as handle_request] (C:\\Note Taking App\\node_modules\\express\\lib\\router\\layer.js:95:5)\n    at trim_prefix (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:328:13)\n    at C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:286:9\n    at Function.process_params (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:346:12)\n    at next (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:280:10)\n    at C:\\Note Taking App\\node_modules\\body-parser\\lib\\read.js:137:5\n    at AsyncResource.runInAsyncScope (node:async_hooks:201:9)\n    at invokeCallback (C:\\Note Taking App\\node_modules\\raw-body\\index.js:238:16)\n    at done (C:\\Note Taking App\\node_modules\\raw-body\\index.js:227:7)\n    at IncomingMessage.onEnd (C:\\Note Taking App\\node_modules\\raw-body\\index.js:287:7)\n    at IncomingMessage.emit (node:events:538:35)\n    at endReadableNT (node:internal/streams/readable:1345:12)\n    at processTicksAndRejections (node:internal/process/task_queues:83:21)","timestamp":"2023-05-22T10:48:39.062Z"}
{"code":422,"data":[{"message":"E-Mail is invalid."}],"level":"error","message":"An error occurred while creating a user: Invalid input.","stack":"Error: Invalid input.\n    at createUser (C:\\Note Taking App\\src\\controllers\\userController.js:30:21)\n    at Layer.handle [as handle_request] (C:\\Note Taking App\\node_modules\\express\\lib\\router\\layer.js:95:5)\n    at next (C:\\Note Taking App\\node_modules\\express\\lib\\router\\route.js:144:13)\n    at Route.dispatch (C:\\Note Taking App\\node_modules\\express\\lib\\router\\route.js:114:3)\n    at Layer.handle [as handle_request] (C:\\Note Taking App\\node_modules\\express\\lib\\router\\layer.js:95:5)\n    at C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:284:15\n    at Function.process_params (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:346:12)\n    at next (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:280:10)\n    at Function.handle (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:175:3)\n    at router (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:47:12)\n    at Layer.handle [as handle_request] (C:\\Note Taking App\\node_modules\\express\\lib\\router\\layer.js:95:5)\n    at trim_prefix (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:328:13)\n    at C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:286:9\n    at Function.process_params (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:346:12)\n    at next (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:280:10)\n    at C:\\Note Taking App\\node_modules\\body-parser\\lib\\read.js:137:5\n    at AsyncResource.runInAsyncScope (node:async_hooks:201:9)\n    at invokeCallback (C:\\Note Taking App\\node_modules\\raw-body\\index.js:238:16)\n    at done (C:\\Note Taking App\\node_modules\\raw-body\\index.js:227:7)\n    at IncomingMessage.onEnd (C:\\Note Taking App\\node_modules\\raw-body\\index.js:287:7)\n    at IncomingMessage.emit (node:events:538:35)\n    at endReadableNT (node:internal/streams/readable:1345:12)\n    at processTicksAndRejections (node:internal/process/task_queues:83:21)","timestamp":"2023-05-22T11:07:39.967Z"}
{"code":422,"data":[{"message":"E-Mail is invalid."}],"level":"error","message":"An error occurred while updating a user: Invalid input.","stack":"Error: Invalid input.\n    at updateUser (C:\\Note Taking App\\src\\controllers\\userController.js:75:21)\n    at Layer.handle [as handle_request] (C:\\Note Taking App\\node_modules\\express\\lib\\router\\layer.js:95:5)\n    at next (C:\\Note Taking App\\node_modules\\express\\lib\\router\\route.js:144:13)\n    at Route.dispatch (C:\\Note Taking App\\node_modules\\express\\lib\\router\\route.js:114:3)\n    at Layer.handle [as handle_request] (C:\\Note Taking App\\node_modules\\express\\lib\\router\\layer.js:95:5)\n    at C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:284:15\n    at param (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:365:14)\n    at param (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:376:14)\n    at Function.process_params (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:421:3)\n    at next (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:280:10)\n    at Function.handle (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:175:3)\n    at router (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:47:12)\n    at Layer.handle [as handle_request] (C:\\Note Taking App\\node_modules\\express\\lib\\router\\layer.js:95:5)\n    at trim_prefix (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:328:13)\n    at C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:286:9\n    at Function.process_params (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:346:12)\n    at next (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:280:10)\n    at C:\\Note Taking App\\node_modules\\body-parser\\lib\\read.js:137:5\n    at AsyncResource.runInAsyncScope (node:async_hooks:201:9)\n    at invokeCallback (C:\\Note Taking App\\node_modules\\raw-body\\index.js:238:16)\n    at done (C:\\Note Taking App\\node_modules\\raw-body\\index.js:227:7)\n    at IncomingMessage.onEnd (C:\\Note Taking App\\node_modules\\raw-body\\index.js:287:7)\n    at IncomingMessage.emit (node:events:538:35)\n    at endReadableNT (node:internal/streams/readable:1345:12)\n    at processTicksAndRejections (node:internal/process/task_queues:83:21)","timestamp":"2023-05-22T11:07:40.210Z"}
{"level":"error","message":"An error occurred while updating a category: password is not defined","stack":"ReferenceError: password is not defined\n    at updateCategory (C:\\Note Taking App\\src\\controllers\\categoryController.js:64:56)\n    at Layer.handle [as handle_request] (C:\\Note Taking App\\node_modules\\express\\lib\\router\\layer.js:95:5)\n    at next (C:\\Note Taking App\\node_modules\\express\\lib\\router\\route.js:144:13)\n    at Route.dispatch (C:\\Note Taking App\\node_modules\\express\\lib\\router\\route.js:114:3)\n    at Layer.handle [as handle_request] (C:\\Note Taking App\\node_modules\\express\\lib\\router\\layer.js:95:5)\n    at C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:284:15\n    at param (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:365:14)\n    at param (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:376:14)\n    at Function.process_params (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:421:3)\n    at next (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:280:10)\n    at Function.handle (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:175:3)\n    at router (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:47:12)\n    at Layer.handle [as handle_request] (C:\\Note Taking App\\node_modules\\express\\lib\\router\\layer.js:95:5)\n    at trim_prefix (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:328:13)\n    at C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:286:9\n    at Function.process_params (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:346:12)\n    at next (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:280:10)\n    at C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:646:15\n    at next (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:265:14)\n    at Function.handle (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:175:3)\n    at router (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:47:12)\n    at Layer.handle [as handle_request] (C:\\Note Taking App\\node_modules\\express\\lib\\router\\layer.js:95:5)\n    at trim_prefix (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:328:13)\n    at C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:286:9\n    at Function.process_params (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:346:12)\n    at next (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:280:10)\n    at C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:646:15\n    at next (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:265:14)\n    at Function.handle (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:175:3)\n    at router (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:47:12)\n    at Layer.handle [as handle_request] (C:\\Note Taking App\\node_modules\\express\\lib\\router\\layer.js:95:5)\n    at trim_prefix (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:328:13)\n    at C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:286:9\n    at Function.process_params (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:346:12)\n    at next (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:280:10)\n    at C:\\Note Taking App\\node_modules\\body-parser\\lib\\read.js:137:5\n    at AsyncResource.runInAsyncScope (node:async_hooks:201:9)\n    at invokeCallback (C:\\Note Taking App\\node_modules\\raw-body\\index.js:238:16)\n    at done (C:\\Note Taking App\\node_modules\\raw-body\\index.js:227:7)\n    at IncomingMessage.onEnd (C:\\Note Taking App\\node_modules\\raw-body\\index.js:287:7)\n    at IncomingMessage.emit (node:events:538:35)\n    at endReadableNT (node:internal/streams/readable:1345:12)\n    at processTicksAndRejections (node:internal/process/task_queues:83:21)","timestamp":"2023-05-22T12:48:59.497Z"}
{"level":"error","message":"An error occurred while deleting a note: Failed to delete the note.","stack":"Error: Failed to delete the note.\n    at NoteManagement.deleteNote (C:\\Note Taking App\\src\\services\\noteServices.js:56:15)\n    at processTicksAndRejections (node:internal/process/task_queues:96:5)\n    at async deleteNote (C:\\Note Taking App\\src\\controllers\\noteController.js:125:25)","timestamp":"2023-05-22T13:32:30.829Z"}
{"level":"error","message":"An error occurred while deleting a note: Failed to delete the note.","stack":"Error: Failed to delete the note.\n    at NoteManagement.deleteNote (C:\\Note Taking App\\src\\services\\noteServices.js:56:15)\n    at processTicksAndRejections (node:internal/process/task_queues:96:5)\n    at async deleteNote (C:\\Note Taking App\\src\\controllers\\noteController.js:125:25)","timestamp":"2023-05-22T13:38:30.822Z"}
{"level":"error","message":"An error occurred while creating a note: utils.isValidObjectID is not a function","stack":"TypeError: utils.isValidObjectID is not a function\n    at createNote (C:\\Note Taking App\\src\\controllers\\noteController.js:39:16)\n    at Layer.handle [as handle_request] (C:\\Note Taking App\\node_modules\\express\\lib\\router\\layer.js:95:5)\n    at next (C:\\Note Taking App\\node_modules\\express\\lib\\router\\route.js:144:13)\n    at Route.dispatch (C:\\Note Taking App\\node_modules\\express\\lib\\router\\route.js:114:3)\n    at Layer.handle [as handle_request] (C:\\Note Taking App\\node_modules\\express\\lib\\router\\layer.js:95:5)\n    at C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:284:15\n    at Function.process_params (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:346:12)\n    at next (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:280:10)\n    at Function.handle (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:175:3)\n    at router (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:47:12)\n    at Layer.handle [as handle_request] (C:\\Note Taking App\\node_modules\\express\\lib\\router\\layer.js:95:5)\n    at trim_prefix (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:328:13)\n    at C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:286:9\n    at Function.process_params (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:346:12)\n    at next (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:280:10)\n    at C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:646:15\n    at next (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:265:14)\n    at Function.handle (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:175:3)\n    at router (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:47:12)\n    at Layer.handle [as handle_request] (C:\\Note Taking App\\node_modules\\express\\lib\\router\\layer.js:95:5)\n    at trim_prefix (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:328:13)\n    at C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:286:9\n    at Function.process_params (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:346:12)\n    at next (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:280:10)\n    at C:\\Note Taking App\\node_modules\\body-parser\\lib\\read.js:137:5\n    at AsyncResource.runInAsyncScope (node:async_hooks:201:9)\n    at invokeCallback (C:\\Note Taking App\\node_modules\\raw-body\\index.js:238:16)\n    at done (C:\\Note Taking App\\node_modules\\raw-body\\index.js:227:7)\n    at IncomingMessage.onEnd (C:\\Note Taking App\\node_modules\\raw-body\\index.js:287:7)\n    at IncomingMessage.emit (node:events:538:35)\n    at endReadableNT (node:internal/streams/readable:1345:12)\n    at processTicksAndRejections (node:internal/process/task_queues:83:21)","timestamp":"2023-05-22T13:52:55.037Z"}
{"level":"error","message":"Error:","timestamp":"2023-05-22T13:52:55.040Z"}
{"level":"error","message":"An error occurred while updating a note: userID is not defined","stack":"ReferenceError: userID is not defined\n    at updateNote (C:\\Note Taking App\\src\\controllers\\noteController.js:87:32)\n    at Layer.handle [as handle_request] (C:\\Note Taking App\\node_modules\\express\\lib\\router\\layer.js:95:5)\n    at next (C:\\Note Taking App\\node_modules\\express\\lib\\router\\route.js:144:13)\n    at Route.dispatch (C:\\Note Taking App\\node_modules\\express\\lib\\router\\route.js:114:3)\n    at Layer.handle [as handle_request] (C:\\Note Taking App\\node_modules\\express\\lib\\router\\layer.js:95:5)\n    at C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:284:15\n    at param (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:365:14)\n    at param (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:376:14)\n    at Function.process_params (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:421:3)\n    at next (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:280:10)\n    at Function.handle (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:175:3)\n    at router (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:47:12)\n    at Layer.handle [as handle_request] (C:\\Note Taking App\\node_modules\\express\\lib\\router\\layer.js:95:5)\n    at trim_prefix (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:328:13)\n    at C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:286:9\n    at Function.process_params (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:346:12)\n    at next (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:280:10)\n    at C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:646:15\n    at next (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:265:14)\n    at Function.handle (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:175:3)\n    at router (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:47:12)\n    at Layer.handle [as handle_request] (C:\\Note Taking App\\node_modules\\express\\lib\\router\\layer.js:95:5)\n    at trim_prefix (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:328:13)\n    at C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:286:9\n    at Function.process_params (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:346:12)\n    at next (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:280:10)\n    at C:\\Note Taking App\\node_modules\\body-parser\\lib\\read.js:137:5\n    at AsyncResource.runInAsyncScope (node:async_hooks:201:9)\n    at invokeCallback (C:\\Note Taking App\\node_modules\\raw-body\\index.js:238:16)\n    at done (C:\\Note Taking App\\node_modules\\raw-body\\index.js:227:7)\n    at IncomingMessage.onEnd (C:\\Note Taking App\\node_modules\\raw-body\\index.js:287:7)\n    at IncomingMessage.emit (node:events:538:35)\n    at endReadableNT (node:internal/streams/readable:1345:12)\n    at processTicksAndRejections (node:internal/process/task_queues:83:21)","timestamp":"2023-05-22T13:52:55.082Z"}
{"level":"error","message":"An error occurred while deleting a note: Failed to delete the note.","stack":"Error: Failed to delete the note.\n    at NoteManagement.deleteNote (C:\\Note Taking App\\src\\services\\noteServices.js:56:15)\n    at processTicksAndRejections (node:internal/process/task_queues:96:5)\n    at async deleteNote (C:\\Note Taking App\\src\\controllers\\noteController.js:125:25)","timestamp":"2023-05-22T13:52:55.118Z"}
{"level":"error","message":"An error occurred while updating a note: title is not defined","stack":"ReferenceError: title is not defined\n    at updateNote (C:\\Note Taking App\\src\\controllers\\noteController.js:95:27)\n    at Layer.handle [as handle_request] (C:\\Note Taking App\\node_modules\\express\\lib\\router\\layer.js:95:5)\n    at next (C:\\Note Taking App\\node_modules\\express\\lib\\router\\route.js:144:13)\n    at Route.dispatch (C:\\Note Taking App\\node_modules\\express\\lib\\router\\route.js:114:3)\n    at Layer.handle [as handle_request] (C:\\Note Taking App\\node_modules\\express\\lib\\router\\layer.js:95:5)\n    at C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:284:15\n    at param (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:365:14)\n    at param (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:376:14)\n    at Function.process_params (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:421:3)\n    at next (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:280:10)\n    at Function.handle (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:175:3)\n    at router (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:47:12)\n    at Layer.handle [as handle_request] (C:\\Note Taking App\\node_modules\\express\\lib\\router\\layer.js:95:5)\n    at trim_prefix (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:328:13)\n    at C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:286:9\n    at Function.process_params (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:346:12)\n    at next (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:280:10)\n    at C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:646:15\n    at next (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:265:14)\n    at Function.handle (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:175:3)\n    at router (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:47:12)\n    at Layer.handle [as handle_request] (C:\\Note Taking App\\node_modules\\express\\lib\\router\\layer.js:95:5)\n    at trim_prefix (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:328:13)\n    at C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:286:9\n    at Function.process_params (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:346:12)\n    at next (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:280:10)\n    at C:\\Note Taking App\\node_modules\\body-parser\\lib\\read.js:137:5\n    at AsyncResource.runInAsyncScope (node:async_hooks:201:9)\n    at invokeCallback (C:\\Note Taking App\\node_modules\\raw-body\\index.js:238:16)\n    at done (C:\\Note Taking App\\node_modules\\raw-body\\index.js:227:7)\n    at IncomingMessage.onEnd (C:\\Note Taking App\\node_modules\\raw-body\\index.js:287:7)\n    at IncomingMessage.emit (node:events:538:35)\n    at endReadableNT (node:internal/streams/readable:1345:12)\n    at processTicksAndRejections (node:internal/process/task_queues:83:21)","timestamp":"2023-05-22T14:18:58.683Z"}
{"level":"error","message":"An error occurred while deleting a note: Failed to delete the note.","stack":"Error: Failed to delete the note.\n    at NoteManagement.deleteNote (C:\\Note Taking App\\src\\services\\noteServices.js:56:15)\n    at processTicksAndRejections (node:internal/process/task_queues:96:5)\n    at async deleteNote (C:\\Note Taking App\\src\\controllers\\noteController.js:125:25)","timestamp":"2023-05-22T14:18:58.726Z"}
{"level":"error","message":"An error occurred while deleting a note: Failed to delete the note.","stack":"Error: Failed to delete the note.\n    at NoteManagement.deleteNote (C:\\Note Taking App\\src\\services\\noteServices.js:56:15)\n    at processTicksAndRejections (node:internal/process/task_queues:96:5)\n    at async deleteNote (C:\\Note Taking App\\src\\controllers\\noteController.js:125:25)","timestamp":"2023-05-22T14:33:57.816Z"}
{"level":"error","message":"An error occurred while updating a note: title is not defined","stack":"ReferenceError: title is not defined\n    at updateNote (C:\\Note Taking App\\src\\controllers\\noteController.js:95:27)\n    at Layer.handle [as handle_request] (C:\\Note Taking App\\node_modules\\express\\lib\\router\\layer.js:95:5)\n    at next (C:\\Note Taking App\\node_modules\\express\\lib\\router\\route.js:144:13)\n    at Route.dispatch (C:\\Note Taking App\\node_modules\\express\\lib\\router\\route.js:114:3)\n    at Layer.handle [as handle_request] (C:\\Note Taking App\\node_modules\\express\\lib\\router\\layer.js:95:5)\n    at C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:284:15\n    at param (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:365:14)\n    at param (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:376:14)\n    at Function.process_params (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:421:3)\n    at next (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:280:10)\n    at Function.handle (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:175:3)\n    at router (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:47:12)\n    at Layer.handle [as handle_request] (C:\\Note Taking App\\node_modules\\express\\lib\\router\\layer.js:95:5)\n    at trim_prefix (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:328:13)\n    at C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:286:9\n    at Function.process_params (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:346:12)\n    at next (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:280:10)\n    at C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:646:15\n    at next (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:265:14)\n    at Function.handle (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:175:3)\n    at router (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:47:12)\n    at Layer.handle [as handle_request] (C:\\Note Taking App\\node_modules\\express\\lib\\router\\layer.js:95:5)\n    at trim_prefix (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:328:13)\n    at C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:286:9\n    at Function.process_params (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:346:12)\n    at next (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:280:10)\n    at C:\\Note Taking App\\node_modules\\body-parser\\lib\\read.js:137:5\n    at AsyncResource.runInAsyncScope (node:async_hooks:201:9)\n    at invokeCallback (C:\\Note Taking App\\node_modules\\raw-body\\index.js:238:16)\n    at done (C:\\Note Taking App\\node_modules\\raw-body\\index.js:227:7)\n    at IncomingMessage.onEnd (C:\\Note Taking App\\node_modules\\raw-body\\index.js:287:7)\n    at IncomingMessage.emit (node:events:538:35)\n    at endReadableNT (node:internal/streams/readable:1345:12)\n    at processTicksAndRejections (node:internal/process/task_queues:83:21)","timestamp":"2023-05-22T14:39:28.900Z"}
{"level":"error","message":"An error occurred while updating a note: title is not defined","stack":"ReferenceError: title is not defined\n    at updateNote (C:\\Note Taking App\\src\\controllers\\noteController.js:95:27)\n    at Layer.handle [as handle_request] (C:\\Note Taking App\\node_modules\\express\\lib\\router\\layer.js:95:5)\n    at next (C:\\Note Taking App\\node_modules\\express\\lib\\router\\route.js:144:13)\n    at Route.dispatch (C:\\Note Taking App\\node_modules\\express\\lib\\router\\route.js:114:3)\n    at Layer.handle [as handle_request] (C:\\Note Taking App\\node_modules\\express\\lib\\router\\layer.js:95:5)\n    at C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:284:15\n    at param (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:365:14)\n    at param (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:376:14)\n    at Function.process_params (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:421:3)\n    at next (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:280:10)\n    at Function.handle (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:175:3)\n    at router (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:47:12)\n    at Layer.handle [as handle_request] (C:\\Note Taking App\\node_modules\\express\\lib\\router\\layer.js:95:5)\n    at trim_prefix (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:328:13)\n    at C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:286:9\n    at Function.process_params (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:346:12)\n    at next (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:280:10)\n    at C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:646:15\n    at next (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:265:14)\n    at Function.handle (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:175:3)\n    at router (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:47:12)\n    at Layer.handle [as handle_request] (C:\\Note Taking App\\node_modules\\express\\lib\\router\\layer.js:95:5)\n    at trim_prefix (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:328:13)\n    at C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:286:9\n    at Function.process_params (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:346:12)\n    at next (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:280:10)\n    at C:\\Note Taking App\\node_modules\\body-parser\\lib\\read.js:137:5\n    at AsyncResource.runInAsyncScope (node:async_hooks:201:9)\n    at invokeCallback (C:\\Note Taking App\\node_modules\\raw-body\\index.js:238:16)\n    at done (C:\\Note Taking App\\node_modules\\raw-body\\index.js:227:7)\n    at IncomingMessage.onEnd (C:\\Note Taking App\\node_modules\\raw-body\\index.js:287:7)\n    at IncomingMessage.emit (node:events:538:35)\n    at endReadableNT (node:internal/streams/readable:1345:12)\n    at processTicksAndRejections (node:internal/process/task_queues:83:21)","timestamp":"2023-05-22T14:46:41.322Z"}
{"level":"error","message":"An error occurred while getting a note by ID: Failed to retrieve the note.","stack":"Error: Failed to retrieve the note.\n    at NoteManagement.getNoteById (C:\\Note Taking App\\src\\services\\noteServices.js:22:15)\n    at processTicksAndRejections (node:internal/process/task_queues:96:5)\n    at async getNoteById (C:\\Note Taking App\\src\\controllers\\noteController.js:21:18)","timestamp":"2023-05-22T15:10:47.137Z"}
{"level":"error","message":"An error occurred while deleting a user: User not found","stack":"Error: User not found\n    at Object.deleteUserStub [as deleteUser] (C:\\Note Taking App\\src\\test\\usersApi.test.js:111:50)\n    at deleteUser (C:\\Note Taking App\\src\\controllers\\userController.js:101:26)\n    at Layer.handle [as handle_request] (C:\\Note Taking App\\node_modules\\express\\lib\\router\\layer.js:95:5)\n    at next (C:\\Note Taking App\\node_modules\\express\\lib\\router\\route.js:144:13)\n    at Route.dispatch (C:\\Note Taking App\\node_modules\\express\\lib\\router\\route.js:114:3)\n    at Layer.handle [as handle_request] (C:\\Note Taking App\\node_modules\\express\\lib\\router\\layer.js:95:5)\n    at C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:284:15\n    at param (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:365:14)\n    at param (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:376:14)\n    at Function.process_params (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:421:3)\n    at next (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:280:10)\n    at Function.handle (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:175:3)\n    at router (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:47:12)\n    at Layer.handle [as handle_request] (C:\\Note Taking App\\node_modules\\express\\lib\\router\\layer.js:95:5)\n    at trim_prefix (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:328:13)\n    at C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:286:9\n    at Function.process_params (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:346:12)\n    at next (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:280:10)\n    at jsonParser (C:\\Note Taking App\\node_modules\\body-parser\\lib\\types\\json.js:113:7)\n    at Layer.handle [as handle_request] (C:\\Note Taking App\\node_modules\\express\\lib\\router\\layer.js:95:5)\n    at trim_prefix (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:328:13)\n    at C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:286:9\n    at Function.process_params (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:346:12)\n    at next (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:280:10)\n    at expressInit (C:\\Note Taking App\\node_modules\\express\\lib\\middleware\\init.js:40:5)\n    at Layer.handle [as handle_request] (C:\\Note Taking App\\node_modules\\express\\lib\\router\\layer.js:95:5)\n    at trim_prefix (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:328:13)\n    at C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:286:9\n    at Function.process_params (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:346:12)\n    at next (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:280:10)\n    at query (C:\\Note Taking App\\node_modules\\express\\lib\\middleware\\query.js:45:5)\n    at Layer.handle [as handle_request] (C:\\Note Taking App\\node_modules\\express\\lib\\router\\layer.js:95:5)\n    at trim_prefix (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:328:13)\n    at C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:286:9\n    at Function.process_params (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:346:12)\n    at next (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:280:10)\n    at Function.handle (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:175:3)\n    at Function.handle (C:\\Note Taking App\\node_modules\\express\\lib\\application.js:181:10)\n    at Server.app (C:\\Note Taking App\\node_modules\\express\\lib\\express.js:39:9)\n    at Server.emit (node:events:526:28)\n    at parserOnIncoming (node:_http_server:951:12)\n    at HTTPParser.parserOnHeadersComplete (node:_http_common:128:17)","timestamp":"2023-05-22T15:42:49.526Z"}
{"level":"error","message":"An error occurred while deleting a user: User not found","stack":"Error: User not found\n    at Object.deleteUserStub [as deleteUser] (C:\\Note Taking App\\src\\test\\usersApi.test.js:111:50)\n    at deleteUser (C:\\Note Taking App\\src\\controllers\\userController.js:101:26)\n    at Layer.handle [as handle_request] (C:\\Note Taking App\\node_modules\\express\\lib\\router\\layer.js:95:5)\n    at next (C:\\Note Taking App\\node_modules\\express\\lib\\router\\route.js:144:13)\n    at Route.dispatch (C:\\Note Taking App\\node_modules\\express\\lib\\router\\route.js:114:3)\n    at Layer.handle [as handle_request] (C:\\Note Taking App\\node_modules\\express\\lib\\router\\layer.js:95:5)\n    at C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:284:15\n    at param (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:365:14)\n    at param (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:376:14)\n    at Function.process_params (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:421:3)\n    at next (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:280:10)\n    at Function.handle (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:175:3)\n    at router (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:47:12)\n    at Layer.handle [as handle_request] (C:\\Note Taking App\\node_modules\\express\\lib\\router\\layer.js:95:5)\n    at trim_prefix (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:328:13)\n    at C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:286:9\n    at Function.process_params (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:346:12)\n    at next (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:280:10)\n    at jsonParser (C:\\Note Taking App\\node_modules\\body-parser\\lib\\types\\json.js:113:7)\n    at Layer.handle [as handle_request] (C:\\Note Taking App\\node_modules\\express\\lib\\router\\layer.js:95:5)\n    at trim_prefix (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:328:13)\n    at C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:286:9\n    at Function.process_params (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:346:12)\n    at next (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:280:10)\n    at expressInit (C:\\Note Taking App\\node_modules\\express\\lib\\middleware\\init.js:40:5)\n    at Layer.handle [as handle_request] (C:\\Note Taking App\\node_modules\\express\\lib\\router\\layer.js:95:5)\n    at trim_prefix (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:328:13)\n    at C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:286:9\n    at Function.process_params (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:346:12)\n    at next (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:280:10)\n    at query (C:\\Note Taking App\\node_modules\\express\\lib\\middleware\\query.js:45:5)\n    at Layer.handle [as handle_request] (C:\\Note Taking App\\node_modules\\express\\lib\\router\\layer.js:95:5)\n    at trim_prefix (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:328:13)\n    at C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:286:9\n    at Function.process_params (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:346:12)\n    at next (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:280:10)\n    at Function.handle (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:175:3)\n    at Function.handle (C:\\Note Taking App\\node_modules\\express\\lib\\application.js:181:10)\n    at Server.app (C:\\Note Taking App\\node_modules\\express\\lib\\express.js:39:9)\n    at Server.emit (node:events:526:28)\n    at parserOnIncoming (node:_http_server:951:12)\n    at HTTPParser.parserOnHeadersComplete (node:_http_common:128:17)","timestamp":"2023-05-22T16:04:08.337Z"}
{"level":"error","message":"An error occurred while creating a user: User is not defined","stack":"ReferenceError: User is not defined\n    at Object.getUserByEmail (C:\\Note Taking App\\src\\utils\\helper.js:11:26)\n    at createUser (C:\\Note Taking App\\src\\controllers\\userController.js:10:38)\n    at Layer.handle [as handle_request] (C:\\Note Taking App\\node_modules\\express\\lib\\router\\layer.js:95:5)\n    at next (C:\\Note Taking App\\node_modules\\express\\lib\\router\\route.js:144:13)\n    at Route.dispatch (C:\\Note Taking App\\node_modules\\express\\lib\\router\\route.js:114:3)\n    at Layer.handle [as handle_request] (C:\\Note Taking App\\node_modules\\express\\lib\\router\\layer.js:95:5)\n    at C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:284:15\n    at Function.process_params (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:346:12)\n    at next (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:280:10)\n    at Function.handle (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:175:3)","timestamp":"2023-05-22T16:22:59.432Z"}
{"level":"error","message":"An error occurred while creating a user: User is not defined","stack":"ReferenceError: User is not defined\n    at Object.getUserByEmail (C:\\Note Taking App\\src\\utils\\helper.js:11:26)\n    at createUser (C:\\Note Taking App\\src\\controllers\\userController.js:10:38)\n    at Layer.handle [as handle_request] (C:\\Note Taking App\\node_modules\\express\\lib\\router\\layer.js:95:5)\n    at next (C:\\Note Taking App\\node_modules\\express\\lib\\router\\route.js:144:13)\n    at Route.dispatch (C:\\Note Taking App\\node_modules\\express\\lib\\router\\route.js:114:3)\n    at Layer.handle [as handle_request] (C:\\Note Taking App\\node_modules\\express\\lib\\router\\layer.js:95:5)\n    at C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:284:15\n    at Function.process_params (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:346:12)\n    at next (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:280:10)\n    at Function.handle (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:175:3)","timestamp":"2023-05-22T16:24:08.127Z"}
{"level":"error","message":"An error occurred while creating a user: User is not defined","stack":"ReferenceError: User is not defined\n    at Object.getUserByEmail (C:\\Note Taking App\\src\\utils\\helper.js:11:26)\n    at createUser (C:\\Note Taking App\\src\\controllers\\userController.js:10:38)\n    at Layer.handle [as handle_request] (C:\\Note Taking App\\node_modules\\express\\lib\\router\\layer.js:95:5)\n    at next (C:\\Note Taking App\\node_modules\\express\\lib\\router\\route.js:144:13)\n    at Route.dispatch (C:\\Note Taking App\\node_modules\\express\\lib\\router\\route.js:114:3)\n    at Layer.handle [as handle_request] (C:\\Note Taking App\\node_modules\\express\\lib\\router\\layer.js:95:5)\n    at C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:284:15\n    at Function.process_params (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:346:12)\n    at next (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:280:10)\n    at Function.handle (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:175:3)","timestamp":"2023-05-22T16:26:32.257Z"}
{"level":"error","message":"An error occurred while creating a user: res.status(...).message is not a function","stack":"TypeError: res.status(...).message is not a function\n    at createUser (C:\\Note Taking App\\src\\controllers\\userController.js:14:24)\n    at processTicksAndRejections (node:internal/process/task_queues:96:5)","timestamp":"2023-05-22T16:42:52.430Z"}
{"level":"error","message":"An error occurred while creating a user: res.status(...).message is not a function","stack":"TypeError: res.status(...).message is not a function\n    at createUser (C:\\Note Taking App\\src\\controllers\\userController.js:14:24)\n    at processTicksAndRejections (node:internal/process/task_queues:96:5)","timestamp":"2023-05-22T16:44:40.512Z"}
{"level":"error","message":"An error occurred while creating a user: res.status(...).message is not a function","stack":"TypeError: res.status(...).message is not a function\n    at createUser (C:\\Note Taking App\\src\\controllers\\userController.js:14:24)\n    at processTicksAndRejections (node:internal/process/task_queues:96:5)","timestamp":"2023-05-22T16:44:43.158Z"}
{"level":"error","message":"An error occurred while creating a user: res.status(...).message is not a function","stack":"TypeError: res.status(...).message is not a function\n    at createUser (C:\\Note Taking App\\src\\controllers\\userController.js:14:24)\n    at processTicksAndRejections (node:internal/process/task_queues:96:5)","timestamp":"2023-05-22T16:44:44.298Z"}
{"level":"error","message":"An error occurred while deleting a user: User not found","stack":"Error: User not found\n    at Object.deleteUserStub [as deleteUser] (C:\\Note Taking App\\src\\test\\usersApi.test.js:111:50)\n    at deleteUser (C:\\Note Taking App\\src\\controllers\\userController.js:102:26)\n    at Layer.handle [as handle_request] (C:\\Note Taking App\\node_modules\\express\\lib\\router\\layer.js:95:5)\n    at next (C:\\Note Taking App\\node_modules\\express\\lib\\router\\route.js:144:13)\n    at Route.dispatch (C:\\Note Taking App\\node_modules\\express\\lib\\router\\route.js:114:3)\n    at Layer.handle [as handle_request] (C:\\Note Taking App\\node_modules\\express\\lib\\router\\layer.js:95:5)\n    at C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:284:15\n    at param (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:365:14)\n    at param (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:376:14)\n    at Function.process_params (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:421:3)\n    at next (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:280:10)\n    at Function.handle (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:175:3)\n    at router (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:47:12)\n    at Layer.handle [as handle_request] (C:\\Note Taking App\\node_modules\\express\\lib\\router\\layer.js:95:5)\n    at trim_prefix (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:328:13)\n    at C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:286:9\n    at Function.process_params (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:346:12)\n    at next (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:280:10)\n    at jsonParser (C:\\Note Taking App\\node_modules\\body-parser\\lib\\types\\json.js:113:7)\n    at Layer.handle [as handle_request] (C:\\Note Taking App\\node_modules\\express\\lib\\router\\layer.js:95:5)\n    at trim_prefix (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:328:13)\n    at C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:286:9\n    at Function.process_params (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:346:12)\n    at next (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:280:10)\n    at expressInit (C:\\Note Taking App\\node_modules\\express\\lib\\middleware\\init.js:40:5)\n    at Layer.handle [as handle_request] (C:\\Note Taking App\\node_modules\\express\\lib\\router\\layer.js:95:5)\n    at trim_prefix (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:328:13)\n    at C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:286:9\n    at Function.process_params (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:346:12)\n    at next (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:280:10)\n    at query (C:\\Note Taking App\\node_modules\\express\\lib\\middleware\\query.js:45:5)\n    at Layer.handle [as handle_request] (C:\\Note Taking App\\node_modules\\express\\lib\\router\\layer.js:95:5)\n    at trim_prefix (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:328:13)\n    at C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:286:9\n    at Function.process_params (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:346:12)\n    at next (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:280:10)\n    at Function.handle (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:175:3)\n    at Function.handle (C:\\Note Taking App\\node_modules\\express\\lib\\application.js:181:10)\n    at Server.app (C:\\Note Taking App\\node_modules\\express\\lib\\express.js:39:9)\n    at Server.emit (node:events:526:28)\n    at parserOnIncoming (node:_http_server:951:12)\n    at HTTPParser.parserOnHeadersComplete (node:_http_common:128:17)","timestamp":"2023-05-22T16:45:47.836Z"}
{"level":"error","message":"An error occurred while deleting a user: User not found","stack":"Error: User not found\n    at Object.deleteUserStub [as deleteUser] (C:\\Note Taking App\\src\\test\\usersApi.test.js:111:50)\n    at deleteUser (C:\\Note Taking App\\src\\controllers\\userController.js:102:26)\n    at Layer.handle [as handle_request] (C:\\Note Taking App\\node_modules\\express\\lib\\router\\layer.js:95:5)\n    at next (C:\\Note Taking App\\node_modules\\express\\lib\\router\\route.js:144:13)\n    at Route.dispatch (C:\\Note Taking App\\node_modules\\express\\lib\\router\\route.js:114:3)\n    at Layer.handle [as handle_request] (C:\\Note Taking App\\node_modules\\express\\lib\\router\\layer.js:95:5)\n    at C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:284:15\n    at param (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:365:14)\n    at param (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:376:14)\n    at Function.process_params (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:421:3)\n    at next (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:280:10)\n    at Function.handle (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:175:3)\n    at router (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:47:12)\n    at Layer.handle [as handle_request] (C:\\Note Taking App\\node_modules\\express\\lib\\router\\layer.js:95:5)\n    at trim_prefix (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:328:13)\n    at C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:286:9\n    at Function.process_params (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:346:12)\n    at next (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:280:10)\n    at jsonParser (C:\\Note Taking App\\node_modules\\body-parser\\lib\\types\\json.js:113:7)\n    at Layer.handle [as handle_request] (C:\\Note Taking App\\node_modules\\express\\lib\\router\\layer.js:95:5)\n    at trim_prefix (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:328:13)\n    at C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:286:9\n    at Function.process_params (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:346:12)\n    at next (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:280:10)\n    at expressInit (C:\\Note Taking App\\node_modules\\express\\lib\\middleware\\init.js:40:5)\n    at Layer.handle [as handle_request] (C:\\Note Taking App\\node_modules\\express\\lib\\router\\layer.js:95:5)\n    at trim_prefix (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:328:13)\n    at C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:286:9\n    at Function.process_params (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:346:12)\n    at next (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:280:10)\n    at query (C:\\Note Taking App\\node_modules\\express\\lib\\middleware\\query.js:45:5)\n    at Layer.handle [as handle_request] (C:\\Note Taking App\\node_modules\\express\\lib\\router\\layer.js:95:5)\n    at trim_prefix (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:328:13)\n    at C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:286:9\n    at Function.process_params (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:346:12)\n    at next (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:280:10)\n    at Function.handle (C:\\Note Taking App\\node_modules\\express\\lib\\router\\index.js:175:3)\n    at Function.handle (C:\\Note Taking App\\node_modules\\express\\lib\\application.js:181:10)\n    at Server.app (C:\\Note Taking App\\node_modules\\express\\lib\\express.js:39:9)\n    at Server.emit (node:events:526:28)\n    at parserOnIncoming (node:_http_server:951:12)\n    at HTTPParser.parserOnHeadersComplete (node:_http_common:128:17)","timestamp":"2023-05-22T16:47:07.121Z"}
{"kind":"ObjectId","level":"error","message":"An error occurred while fetching user by ID: Cast to ObjectId failed for value \"123\" (type string) at path \"_id\" for model \"User\"","path":"_id","reason":{},"stack":"CastError: Cast to ObjectId failed for value \"123\" (type string) at path \"_id\" for model \"User\"\n    at ObjectId.cast (C:\\Note Taking App\\node_modules\\mongoose\\lib\\schema\\objectid.js:248:11)\n    at ObjectId.SchemaType.applySetters (C:\\Note Taking App\\node_modules\\mongoose\\lib\\schematype.js:1202:12)\n    at ObjectId.SchemaType.castForQuery (C:\\Note Taking App\\node_modules\\mongoose\\lib\\schematype.js:1614:15)\n    at cast (C:\\Note Taking App\\node_modules\\mongoose\\lib\\cast.js:348:32)\n    at model.Query.Query.cast (C:\\Note Taking App\\node_modules\\mongoose\\lib\\query.js:4819:12)\n    at model.Query.Query._castConditions (C:\\Note Taking App\\node_modules\\mongoose\\lib\\query.js:2193:10)\n    at model.Query._findOne (C:\\Note Taking App\\node_modules\\mongoose\\lib\\query.js:2477:8)\n    at model.Query.exec (C:\\Note Taking App\\node_modules\\mongoose\\lib\\query.js:4349:28)\n    at processTicksAndRejections (node:internal/process/task_queues:96:5)\n    at async UserManagement.getUserById (C:\\Note Taking App\\src\\services\\userServices.js:11:18)\n    at async getUserById (C:\\Note Taking App\\src\\controllers\\userController.js:51:18)","stringValue":"\"123\"","timestamp":"2023-05-22T17:30:47.636Z","value":"123","valueType":"string"}
{"kind":"ObjectId","level":"error","message":"An error occurred while fetching user by ID: Cast to ObjectId failed for value \"123\" (type string) at path \"_id\" for model \"User\"","path":"_id","reason":{},"stack":"CastError: Cast to ObjectId failed for value \"123\" (type string) at path \"_id\" for model \"User\"\n    at ObjectId.cast (C:\\Note Taking App\\node_modules\\mongoose\\lib\\schema\\objectid.js:248:11)\n    at ObjectId.SchemaType.applySetters (C:\\Note Taking App\\node_modules\\mongoose\\lib\\schematype.js:1202:12)\n    at ObjectId.SchemaType.castForQuery (C:\\Note Taking App\\node_modules\\mongoose\\lib\\schematype.js:1614:15)\n    at cast (C:\\Note Taking App\\node_modules\\mongoose\\lib\\cast.js:348:32)\n    at model.Query.Query.cast (C:\\Note Taking App\\node_modules\\mongoose\\lib\\query.js:4819:12)\n    at model.Query.Query._castConditions (C:\\Note Taking App\\node_modules\\mongoose\\lib\\query.js:2193:10)\n    at model.Query._findOne (C:\\Note Taking App\\node_modules\\mongoose\\lib\\query.js:2477:8)\n    at model.Query.exec (C:\\Note Taking App\\node_modules\\mongoose\\lib\\query.js:4349:28)\n    at processTicksAndRejections (node:internal/process/task_queues:96:5)\n    at async UserManagement.getUserById (C:\\Note Taking App\\src\\services\\userServices.js:11:18)\n    at async getUserById (C:\\Note Taking App\\src\\controllers\\userController.js:51:18)","stringValue":"\"123\"","timestamp":"2023-05-22T17:30:47.674Z","value":"123","valueType":"string"}
{"kind":"ObjectId","level":"error","message":"An error occurred while updating a user: Cast to ObjectId failed for value \"123\" (type string) at path \"_id\" for model \"User\"","path":"_id","reason":{},"stack":"CastError: Cast to ObjectId failed for value \"123\" (type string) at path \"_id\" for model \"User\"\n    at ObjectId.cast (C:\\Note Taking App\\node_modules\\mongoose\\lib\\schema\\objectid.js:248:11)\n    at ObjectId.SchemaType.applySetters (C:\\Note Taking App\\node_modules\\mongoose\\lib\\schematype.js:1202:12)\n    at ObjectId.SchemaType.castForQuery (C:\\Note Taking App\\node_modules\\mongoose\\lib\\schematype.js:1614:15)\n    at cast (C:\\Note Taking App\\node_modules\\mongoose\\lib\\cast.js:348:32)\n    at model.Query.Query.cast (C:\\Note Taking App\\node_modules\\mongoose\\lib\\query.js:4819:12)\n    at model.Query.Query._castConditions (C:\\Note Taking App\\node_modules\\mongoose\\lib\\query.js:2193:10)\n    at model.Query._findOneAndUpdate (C:\\Note Taking App\\node_modules\\mongoose\\lib\\query.js:3223:8)\n    at model.Query.exec (C:\\Note Taking App\\node_modules\\mongoose\\lib\\query.js:4349:28)\n    at processTicksAndRejections (node:internal/process/task_queues:96:5)\n    at async UserManagement.updateUser (C:\\Note Taking App\\src\\services\\userServices.js:17:18)\n    at async updateUser (C:\\Note Taking App\\src\\controllers\\userController.js:86:25)","stringValue":"\"123\"","timestamp":"2023-05-22T17:30:47.722Z","value":"123","valueType":"string"}
{"kind":"ObjectId","level":"error","message":"An error occurred while updating a user: Cast to ObjectId failed for value \"123\" (type string) at path \"_id\" for model \"User\"","path":"_id","reason":{},"stack":"CastError: Cast to ObjectId failed for value \"123\" (type string) at path \"_id\" for model \"User\"\n    at ObjectId.cast (C:\\Note Taking App\\node_modules\\mongoose\\lib\\schema\\objectid.js:248:11)\n    at ObjectId.SchemaType.applySetters (C:\\Note Taking App\\node_modules\\mongoose\\lib\\schematype.js:1202:12)\n    at ObjectId.SchemaType.castForQuery (C:\\Note Taking App\\node_modules\\mongoose\\lib\\schematype.js:1614:15)\n    at cast (C:\\Note Taking App\\node_modules\\mongoose\\lib\\cast.js:348:32)\n    at model.Query.Query.cast (C:\\Note Taking App\\node_modules\\mongoose\\lib\\query.js:4819:12)\n    at model.Query.Query._castConditions (C:\\Note Taking App\\node_modules\\mongoose\\lib\\query.js:2193:10)\n    at model.Query._findOneAndUpdate (C:\\Note Taking App\\node_modules\\mongoose\\lib\\query.js:3223:8)\n    at model.Query.exec (C:\\Note Taking App\\node_modules\\mongoose\\lib\\query.js:4349:28)\n    at processTicksAndRejections (node:internal/process/task_queues:96:5)\n    at async UserManagement.updateUser (C:\\Note Taking App\\src\\services\\userServices.js:17:18)\n    at async updateUser (C:\\Note Taking App\\src\\controllers\\userController.js:86:25)","stringValue":"\"123\"","timestamp":"2023-05-22T17:30:47.746Z","value":"123","valueType":"string"}
{"kind":"ObjectId","level":"error","message":"An error occurred while deleting a user: Cast to ObjectId failed for value \"123\" (type string) at path \"_id\" for model \"User\"","path":"_id","reason":{},"stack":"CastError: Cast to ObjectId failed for value \"123\" (type string) at path \"_id\" for model \"User\"\n    at ObjectId.cast (C:\\Note Taking App\\node_modules\\mongoose\\lib\\schema\\objectid.js:248:11)\n    at ObjectId.SchemaType.applySetters (C:\\Note Taking App\\node_modules\\mongoose\\lib\\schematype.js:1202:12)\n    at ObjectId.SchemaType.castForQuery (C:\\Note Taking App\\node_modules\\mongoose\\lib\\schematype.js:1614:15)\n    at cast (C:\\Note Taking App\\node_modules\\mongoose\\lib\\cast.js:348:32)\n    at model.Query.Query.cast (C:\\Note Taking App\\node_modules\\mongoose\\lib\\query.js:4819:12)\n    at model.Query.Query._castConditions (C:\\Note Taking App\\node_modules\\mongoose\\lib\\query.js:2193:10)\n    at model.Query._findOneAndDelete (C:\\Note Taking App\\node_modules\\mongoose\\lib\\query.js:3422:8)\n    at model.Query._findOneAndRemove (C:\\Note Taking App\\node_modules\\mongoose\\lib\\query.js:3650:15)\n    at model.Query.exec (C:\\Note Taking App\\node_modules\\mongoose\\lib\\query.js:4349:28)\n    at processTicksAndRejections (node:internal/process/task_queues:96:5)\n    at async UserManagement.deleteUser (C:\\Note Taking App\\src\\services\\userServices.js:22:5)\n    at async deleteUser (C:\\Note Taking App\\src\\controllers\\userController.js:106:5)","stringValue":"\"123\"","timestamp":"2023-05-22T17:30:47.788Z","value":"123","valueType":"string"}
{"kind":"ObjectId","level":"error","message":"An error occurred while deleting a user: Cast to ObjectId failed for value \"123\" (type string) at path \"_id\" for model \"User\"","path":"_id","reason":{},"stack":"CastError: Cast to ObjectId failed for value \"123\" (type string) at path \"_id\" for model \"User\"\n    at ObjectId.cast (C:\\Note Taking App\\node_modules\\mongoose\\lib\\schema\\objectid.js:248:11)\n    at ObjectId.SchemaType.applySetters (C:\\Note Taking App\\node_modules\\mongoose\\lib\\schematype.js:1202:12)\n    at ObjectId.SchemaType.castForQuery (C:\\Note Taking App\\node_modules\\mongoose\\lib\\schematype.js:1614:15)\n    at cast (C:\\Note Taking App\\node_modules\\mongoose\\lib\\cast.js:348:32)\n    at model.Query.Query.cast (C:\\Note Taking App\\node_modules\\mongoose\\lib\\query.js:4819:12)\n    at model.Query.Query._castConditions (C:\\Note Taking App\\node_modules\\mongoose\\lib\\query.js:2193:10)\n    at model.Query._findOneAndDelete (C:\\Note Taking App\\node_modules\\mongoose\\lib\\query.js:3422:8)\n    at model.Query._findOneAndRemove (C:\\Note Taking App\\node_modules\\mongoose\\lib\\query.js:3650:15)\n    at model.Query.exec (C:\\Note Taking App\\node_modules\\mongoose\\lib\\query.js:4349:28)\n    at processTicksAndRejections (node:internal/process/task_queues:96:5)\n    at async UserManagement.deleteUser (C:\\Note Taking App\\src\\services\\userServices.js:22:5)\n    at async deleteUser (C:\\Note Taking App\\src\\controllers\\userController.js:106:5)","stringValue":"\"123\"","timestamp":"2023-05-22T17:30:47.813Z","value":"123","valueType":"string"}
{"kind":"ObjectId","level":"error","message":"An error occurred while fetching user by ID: Cast to ObjectId failed for value \"123\" (type string) at path \"_id\" for model \"User\"","path":"_id","reason":{},"stack":"CastError: Cast to ObjectId failed for value \"123\" (type string) at path \"_id\" for model \"User\"\n    at ObjectId.cast (C:\\Note Taking App\\node_modules\\mongoose\\lib\\schema\\objectid.js:248:11)\n    at ObjectId.SchemaType.applySetters (C:\\Note Taking App\\node_modules\\mongoose\\lib\\schematype.js:1202:12)\n    at ObjectId.SchemaType.castForQuery (C:\\Note Taking App\\node_modules\\mongoose\\lib\\schematype.js:1614:15)\n    at cast (C:\\Note Taking App\\node_modules\\mongoose\\lib\\cast.js:348:32)\n    at model.Query.Query.cast (C:\\Note Taking App\\node_modules\\mongoose\\lib\\query.js:4819:12)\n    at model.Query.Query._castConditions (C:\\Note Taking App\\node_modules\\mongoose\\lib\\query.js:2193:10)\n    at model.Query._findOne (C:\\Note Taking App\\node_modules\\mongoose\\lib\\query.js:2477:8)\n    at model.Query.exec (C:\\Note Taking App\\node_modules\\mongoose\\lib\\query.js:4349:28)\n    at processTicksAndRejections (node:internal/process/task_queues:96:5)\n    at async UserManagement.getUserById (C:\\Note Taking App\\src\\services\\userServices.js:11:18)\n    at async getUserById (C:\\Note Taking App\\src\\controllers\\userController.js:51:18)","stringValue":"\"123\"","timestamp":"2023-05-22T17:33:49.375Z","value":"123","valueType":"string"}
{"kind":"ObjectId","level":"error","message":"An error occurred while fetching user by ID: Cast to ObjectId failed for value \"123\" (type string) at path \"_id\" for model \"User\"","path":"_id","reason":{},"stack":"CastError: Cast to ObjectId failed for value \"123\" (type string) at path \"_id\" for model \"User\"\n    at ObjectId.cast (C:\\Note Taking App\\node_modules\\mongoose\\lib\\schema\\objectid.js:248:11)\n    at ObjectId.SchemaType.applySetters (C:\\Note Taking App\\node_modules\\mongoose\\lib\\schematype.js:1202:12)\n    at ObjectId.SchemaType.castForQuery (C:\\Note Taking App\\node_modules\\mongoose\\lib\\schematype.js:1614:15)\n    at cast (C:\\Note Taking App\\node_modules\\mongoose\\lib\\cast.js:348:32)\n    at model.Query.Query.cast (C:\\Note Taking App\\node_modules\\mongoose\\lib\\query.js:4819:12)\n    at model.Query.Query._castConditions (C:\\Note Taking App\\node_modules\\mongoose\\lib\\query.js:2193:10)\n    at model.Query._findOne (C:\\Note Taking App\\node_modules\\mongoose\\lib\\query.js:2477:8)\n    at model.Query.exec (C:\\Note Taking App\\node_modules\\mongoose\\lib\\query.js:4349:28)\n    at processTicksAndRejections (node:internal/process/task_queues:96:5)\n    at async UserManagement.getUserById (C:\\Note Taking App\\src\\services\\userServices.js:11:18)\n    at async getUserById (C:\\Note Taking App\\src\\controllers\\userController.js:51:18)","stringValue":"\"123\"","timestamp":"2023-05-22T17:33:49.407Z","value":"123","valueType":"string"}
{"kind":"ObjectId","level":"error","message":"An error occurred while updating a user: Cast to ObjectId failed for value \"123\" (type string) at path \"_id\" for model \"User\"","path":"_id","reason":{},"stack":"CastError: Cast to ObjectId failed for value \"123\" (type string) at path \"_id\" for model \"User\"\n    at ObjectId.cast (C:\\Note Taking App\\node_modules\\mongoose\\lib\\schema\\objectid.js:248:11)\n    at ObjectId.SchemaType.applySetters (C:\\Note Taking App\\node_modules\\mongoose\\lib\\schematype.js:1202:12)\n    at ObjectId.SchemaType.castForQuery (C:\\Note Taking App\\node_modules\\mongoose\\lib\\schematype.js:1614:15)\n    at cast (C:\\Note Taking App\\node_modules\\mongoose\\lib\\cast.js:348:32)\n    at model.Query.Query.cast (C:\\Note Taking App\\node_modules\\mongoose\\lib\\query.js:4819:12)\n    at model.Query.Query._castConditions (C:\\Note Taking App\\node_modules\\mongoose\\lib\\query.js:2193:10)\n    at model.Query._findOneAndUpdate (C:\\Note Taking App\\node_modules\\mongoose\\lib\\query.js:3223:8)\n    at model.Query.exec (C:\\Note Taking App\\node_modules\\mongoose\\lib\\query.js:4349:28)\n    at processTicksAndRejections (node:internal/process/task_queues:96:5)\n    at async UserManagement.updateUser (C:\\Note Taking App\\src\\services\\userServices.js:17:18)\n    at async updateUser (C:\\Note Taking App\\src\\controllers\\userController.js:86:25)","stringValue":"\"123\"","timestamp":"2023-05-22T17:33:49.450Z","value":"123","valueType":"string"}
{"kind":"ObjectId","level":"error","message":"An error occurred while updating a user: Cast to ObjectId failed for value \"123\" (type string) at path \"_id\" for model \"User\"","path":"_id","reason":{},"stack":"CastError: Cast to ObjectId failed for value \"123\" (type string) at path \"_id\" for model \"User\"\n    at ObjectId.cast (C:\\Note Taking App\\node_modules\\mongoose\\lib\\schema\\objectid.js:248:11)\n    at ObjectId.SchemaType.applySetters (C:\\Note Taking App\\node_modules\\mongoose\\lib\\schematype.js:1202:12)\n    at ObjectId.SchemaType.castForQuery (C:\\Note Taking App\\node_modules\\mongoose\\lib\\schematype.js:1614:15)\n    at cast (C:\\Note Taking App\\node_modules\\mongoose\\lib\\cast.js:348:32)\n    at model.Query.Query.cast (C:\\Note Taking App\\node_modules\\mongoose\\lib\\query.js:4819:12)\n    at model.Query.Query._castConditions (C:\\Note Taking App\\node_modules\\mongoose\\lib\\query.js:2193:10)\n    at model.Query._findOneAndUpdate (C:\\Note Taking App\\node_modules\\mongoose\\lib\\query.js:3223:8)\n    at model.Query.exec (C:\\Note Taking App\\node_modules\\mongoose\\lib\\query.js:4349:28)\n    at processTicksAndRejections (node:internal/process/task_queues:96:5)\n    at async UserManagement.updateUser (C:\\Note Taking App\\src\\services\\userServices.js:17:18)\n    at async updateUser (C:\\Note Taking App\\src\\controllers\\userController.js:86:25)","stringValue":"\"123\"","timestamp":"2023-05-22T17:33:49.481Z","value":"123","valueType":"string"}
{"kind":"ObjectId","level":"error","message":"An error occurred while deleting a user: Cast to ObjectId failed for value \"123\" (type string) at path \"_id\" for model \"User\"","path":"_id","reason":{},"stack":"CastError: Cast to ObjectId failed for value \"123\" (type string) at path \"_id\" for model \"User\"\n    at ObjectId.cast (C:\\Note Taking App\\node_modules\\mongoose\\lib\\schema\\objectid.js:248:11)\n    at ObjectId.SchemaType.applySetters (C:\\Note Taking App\\node_modules\\mongoose\\lib\\schematype.js:1202:12)\n    at ObjectId.SchemaType.castForQuery (C:\\Note Taking App\\node_modules\\mongoose\\lib\\schematype.js:1614:15)\n    at cast (C:\\Note Taking App\\node_modules\\mongoose\\lib\\cast.js:348:32)\n    at model.Query.Query.cast (C:\\Note Taking App\\node_modules\\mongoose\\lib\\query.js:4819:12)\n    at model.Query.Query._castConditions (C:\\Note Taking App\\node_modules\\mongoose\\lib\\query.js:2193:10)\n    at model.Query._findOneAndDelete (C:\\Note Taking App\\node_modules\\mongoose\\lib\\query.js:3422:8)\n    at model.Query._findOneAndRemove (C:\\Note Taking App\\node_modules\\mongoose\\lib\\query.js:3650:15)\n    at model.Query.exec (C:\\Note Taking App\\node_modules\\mongoose\\lib\\query.js:4349:28)\n    at processTicksAndRejections (node:internal/process/task_queues:96:5)\n    at async UserManagement.deleteUser (C:\\Note Taking App\\src\\services\\userServices.js:22:5)\n    at async deleteUser (C:\\Note Taking App\\src\\controllers\\userController.js:106:5)","stringValue":"\"123\"","timestamp":"2023-05-22T17:33:49.507Z","value":"123","valueType":"string"}
{"kind":"ObjectId","level":"error","message":"An error occurred while deleting a user: Cast to ObjectId failed for value \"123\" (type string) at path \"_id\" for model \"User\"","path":"_id","reason":{},"stack":"CastError: Cast to ObjectId failed for value \"123\" (type string) at path \"_id\" for model \"User\"\n    at ObjectId.cast (C:\\Note Taking App\\node_modules\\mongoose\\lib\\schema\\objectid.js:248:11)\n    at ObjectId.SchemaType.applySetters (C:\\Note Taking App\\node_modules\\mongoose\\lib\\schematype.js:1202:12)\n    at ObjectId.SchemaType.castForQuery (C:\\Note Taking App\\node_modules\\mongoose\\lib\\schematype.js:1614:15)\n    at cast (C:\\Note Taking App\\node_modules\\mongoose\\lib\\cast.js:348:32)\n    at model.Query.Query.cast (C:\\Note Taking App\\node_modules\\mongoose\\lib\\query.js:4819:12)\n    at model.Query.Query._castConditions (C:\\Note Taking App\\node_modules\\mongoose\\lib\\query.js:2193:10)\n    at model.Query._findOneAndDelete (C:\\Note Taking App\\node_modules\\mongoose\\lib\\query.js:3422:8)\n    at model.Query._findOneAndRemove (C:\\Note Taking App\\node_modules\\mongoose\\lib\\query.js:3650:15)\n    at model.Query.exec (C:\\Note Taking App\\node_modules\\mongoose\\lib\\query.js:4349:28)\n    at processTicksAndRejections (node:internal/process/task_queues:96:5)\n    at async UserManagement.deleteUser (C:\\Note Taking App\\src\\services\\userServices.js:22:5)\n    at async deleteUser (C:\\Note Taking App\\src\\controllers\\userController.js:106:5)","stringValue":"\"123\"","timestamp":"2023-05-22T17:33:49.536Z","value":"123","valueType":"string"}
{"level":"error","message":"An error occurred while creating a user: res.status(...).message is not a function","stack":"TypeError: res.status(...).message is not a function\n    at createUser (C:\\Note Taking App\\src\\controllers\\userController.js:18:24)\n    at processTicksAndRejections (node:internal/process/task_queues:96:5)","timestamp":"2023-05-22T17:46:17.852Z"}
{"kind":"ObjectId","level":"error","message":"An error occurred while fetching user by ID: Cast to ObjectId failed for value \"123\" (type string) at path \"_id\" for model \"User\"","path":"_id","reason":{},"stack":"CastError: Cast to ObjectId failed for value \"123\" (type string) at path \"_id\" for model \"User\"\n    at ObjectId.cast (C:\\Note Taking App\\node_modules\\mongoose\\lib\\schema\\objectid.js:248:11)\n    at ObjectId.SchemaType.applySetters (C:\\Note Taking App\\node_modules\\mongoose\\lib\\schematype.js:1202:12)\n    at ObjectId.SchemaType.castForQuery (C:\\Note Taking App\\node_modules\\mongoose\\lib\\schematype.js:1614:15)\n    at cast (C:\\Note Taking App\\node_modules\\mongoose\\lib\\cast.js:348:32)\n    at model.Query.Query.cast (C:\\Note Taking App\\node_modules\\mongoose\\lib\\query.js:4819:12)\n    at model.Query.Query._castConditions (C:\\Note Taking App\\node_modules\\mongoose\\lib\\query.js:2193:10)\n    at model.Query._findOne (C:\\Note Taking App\\node_modules\\mongoose\\lib\\query.js:2477:8)\n    at model.Query.exec (C:\\Note Taking App\\node_modules\\mongoose\\lib\\query.js:4349:28)\n    at processTicksAndRejections (node:internal/process/task_queues:96:5)\n    at async UserManagement.getUserById (C:\\Note Taking App\\src\\services\\userServices.js:11:18)\n    at async getUserById (C:\\Note Taking App\\src\\controllers\\userController.js:51:18)","stringValue":"\"123\"","timestamp":"2023-05-22T17:46:17.913Z","value":"123","valueType":"string"}
{"kind":"ObjectId","level":"error","message":"An error occurred while fetching user by ID: Cast to ObjectId failed for value \"123\" (type string) at path \"_id\" for model \"User\"","path":"_id","reason":{},"stack":"CastError: Cast to ObjectId failed for value \"123\" (type string) at path \"_id\" for model \"User\"\n    at ObjectId.cast (C:\\Note Taking App\\node_modules\\mongoose\\lib\\schema\\objectid.js:248:11)\n    at ObjectId.SchemaType.applySetters (C:\\Note Taking App\\node_modules\\mongoose\\lib\\schematype.js:1202:12)\n    at ObjectId.SchemaType.castForQuery (C:\\Note Taking App\\node_modules\\mongoose\\lib\\schematype.js:1614:15)\n    at cast (C:\\Note Taking App\\node_modules\\mongoose\\lib\\cast.js:348:32)\n    at model.Query.Query.cast (C:\\Note Taking App\\node_modules\\mongoose\\lib\\query.js:4819:12)\n    at model.Query.Query._castConditions (C:\\Note Taking App\\node_modules\\mongoose\\lib\\query.js:2193:10)\n    at model.Query._findOne (C:\\Note Taking App\\node_modules\\mongoose\\lib\\query.js:2477:8)\n    at model.Query.exec (C:\\Note Taking App\\node_modules\\mongoose\\lib\\query.js:4349:28)\n    at processTicksAndRejections (node:internal/process/task_queues:96:5)\n    at async UserManagement.getUserById (C:\\Note Taking App\\src\\services\\userServices.js:11:18)\n    at async getUserById (C:\\Note Taking App\\src\\controllers\\userController.js:51:18)","stringValue":"\"123\"","timestamp":"2023-05-22T17:46:17.941Z","value":"123","valueType":"string"}
{"kind":"ObjectId","level":"error","message":"An error occurred while updating a user: Cast to ObjectId failed for value \"123\" (type string) at path \"_id\" for model \"User\"","path":"_id","reason":{},"stack":"CastError: Cast to ObjectId failed for value \"123\" (type string) at path \"_id\" for model \"User\"\n    at ObjectId.cast (C:\\Note Taking App\\node_modules\\mongoose\\lib\\schema\\objectid.js:248:11)\n    at ObjectId.SchemaType.applySetters (C:\\Note Taking App\\node_modules\\mongoose\\lib\\schematype.js:1202:12)\n    at ObjectId.SchemaType.castForQuery (C:\\Note Taking App\\node_modules\\mongoose\\lib\\schematype.js:1614:15)\n    at cast (C:\\Note Taking App\\node_modules\\mongoose\\lib\\cast.js:348:32)\n    at model.Query.Query.cast (C:\\Note Taking App\\node_modules\\mongoose\\lib\\query.js:4819:12)\n    at model.Query.Query._castConditions (C:\\Note Taking App\\node_modules\\mongoose\\lib\\query.js:2193:10)\n    at model.Query._findOneAndUpdate (C:\\Note Taking App\\node_modules\\mongoose\\lib\\query.js:3223:8)\n    at model.Query.exec (C:\\Note Taking App\\node_modules\\mongoose\\lib\\query.js:4349:28)\n    at processTicksAndRejections (node:internal/process/task_queues:96:5)\n    at async UserManagement.updateUser (C:\\Note Taking App\\src\\services\\userServices.js:17:18)\n    at async updateUser (C:\\Note Taking App\\src\\controllers\\userController.js:86:25)","stringValue":"\"123\"","timestamp":"2023-05-22T17:46:17.987Z","value":"123","valueType":"string"}
{"kind":"ObjectId","level":"error","message":"An error occurred while updating a user: Cast to ObjectId failed for value \"123\" (type string) at path \"_id\" for model \"User\"","path":"_id","reason":{},"stack":"CastError: Cast to ObjectId failed for value \"123\" (type string) at path \"_id\" for model \"User\"\n    at ObjectId.cast (C:\\Note Taking App\\node_modules\\mongoose\\lib\\schema\\objectid.js:248:11)\n    at ObjectId.SchemaType.applySetters (C:\\Note Taking App\\node_modules\\mongoose\\lib\\schematype.js:1202:12)\n    at ObjectId.SchemaType.castForQuery (C:\\Note Taking App\\node_modules\\mongoose\\lib\\schematype.js:1614:15)\n    at cast (C:\\Note Taking App\\node_modules\\mongoose\\lib\\cast.js:348:32)\n    at model.Query.Query.cast (C:\\Note Taking App\\node_modules\\mongoose\\lib\\query.js:4819:12)\n    at model.Query.Query._castConditions (C:\\Note Taking App\\node_modules\\mongoose\\lib\\query.js:2193:10)\n    at model.Query._findOneAndUpdate (C:\\Note Taking App\\node_modules\\mongoose\\lib\\query.js:3223:8)\n    at model.Query.exec (C:\\Note Taking App\\node_modules\\mongoose\\lib\\query.js:4349:28)\n    at processTicksAndRejections (node:internal/process/task_queues:96:5)\n    at async UserManagement.updateUser (C:\\Note Taking App\\src\\services\\userServices.js:17:18)\n    at async updateUser (C:\\Note Taking App\\src\\controllers\\userController.js:86:25)","stringValue":"\"123\"","timestamp":"2023-05-22T17:46:18.026Z","value":"123","valueType":"string"}
{"kind":"ObjectId","level":"error","message":"An error occurred while deleting a user: Cast to ObjectId failed for value \"123\" (type string) at path \"_id\" for model \"User\"","path":"_id","reason":{},"stack":"CastError: Cast to ObjectId failed for value \"123\" (type string) at path \"_id\" for model \"User\"\n    at ObjectId.cast (C:\\Note Taking App\\node_modules\\mongoose\\lib\\schema\\objectid.js:248:11)\n    at ObjectId.SchemaType.applySetters (C:\\Note Taking App\\node_modules\\mongoose\\lib\\schematype.js:1202:12)\n    at ObjectId.SchemaType.castForQuery (C:\\Note Taking App\\node_modules\\mongoose\\lib\\schematype.js:1614:15)\n    at cast (C:\\Note Taking App\\node_modules\\mongoose\\lib\\cast.js:348:32)\n    at model.Query.Query.cast (C:\\Note Taking App\\node_modules\\mongoose\\lib\\query.js:4819:12)\n    at model.Query.Query._castConditions (C:\\Note Taking App\\node_modules\\mongoose\\lib\\query.js:2193:10)\n    at model.Query._findOneAndDelete (C:\\Note Taking App\\node_modules\\mongoose\\lib\\query.js:3422:8)\n    at model.Query._findOneAndRemove (C:\\Note Taking App\\node_modules\\mongoose\\lib\\query.js:3650:15)\n    at model.Query.exec (C:\\Note Taking App\\node_modules\\mongoose\\lib\\query.js:4349:28)\n    at processTicksAndRejections (node:internal/process/task_queues:96:5)\n    at async UserManagement.deleteUser (C:\\Note Taking App\\src\\services\\userServices.js:22:5)\n    at async deleteUser (C:\\Note Taking App\\src\\controllers\\userController.js:106:5)","stringValue":"\"123\"","timestamp":"2023-05-22T17:46:18.057Z","value":"123","valueType":"string"}
{"kind":"ObjectId","level":"error","message":"An error occurred while deleting a user: Cast to ObjectId failed for value \"123\" (type string) at path \"_id\" for model \"User\"","path":"_id","reason":{},"stack":"CastError: Cast to ObjectId failed for value \"123\" (type string) at path \"_id\" for model \"User\"\n    at ObjectId.cast (C:\\Note Taking App\\node_modules\\mongoose\\lib\\schema\\objectid.js:248:11)\n    at ObjectId.SchemaType.applySetters (C:\\Note Taking App\\node_modules\\mongoose\\lib\\schematype.js:1202:12)\n    at ObjectId.SchemaType.castForQuery (C:\\Note Taking App\\node_modules\\mongoose\\lib\\schematype.js:1614:15)\n    at cast (C:\\Note Taking App\\node_modules\\mongoose\\lib\\cast.js:348:32)\n    at model.Query.Query.cast (C:\\Note Taking App\\node_modules\\mongoose\\lib\\query.js:4819:12)\n    at model.Query.Query._castConditions (C:\\Note Taking App\\node_modules\\mongoose\\lib\\query.js:2193:10)\n    at model.Query._findOneAndDelete (C:\\Note Taking App\\node_modules\\mongoose\\lib\\query.js:3422:8)\n    at model.Query._findOneAndRemove (C:\\Note Taking App\\node_modules\\mongoose\\lib\\query.js:3650:15)\n    at model.Query.exec (C:\\Note Taking App\\node_modules\\mongoose\\lib\\query.js:4349:28)\n    at processTicksAndRejections (node:internal/process/task_queues:96:5)\n    at async UserManagement.deleteUser (C:\\Note Taking App\\src\\services\\userServices.js:22:5)\n    at async deleteUser (C:\\Note Taking App\\src\\controllers\\userController.js:106:5)","stringValue":"\"123\"","timestamp":"2023-05-22T17:46:18.081Z","value":"123","valueType":"string"}
{"level":"error","message":"An error occurred while creating a user: res.status(...).message is not a function","stack":"TypeError: res.status(...).message is not a function\n    at createUser (C:\\Note Taking App\\src\\controllers\\userController.js:18:24)\n    at processTicksAndRejections (node:internal/process/task_queues:96:5)","timestamp":"2023-05-22T17:53:35.517Z"}
{"kind":"ObjectId","level":"error","message":"An error occurred while fetching user by ID: Cast to ObjectId failed for value \"123\" (type string) at path \"_id\" for model \"User\"","path":"_id","reason":{},"stack":"CastError: Cast to ObjectId failed for value \"123\" (type string) at path \"_id\" for model \"User\"\n    at ObjectId.cast (C:\\Note Taking App\\node_modules\\mongoose\\lib\\schema\\objectid.js:248:11)\n    at ObjectId.SchemaType.applySetters (C:\\Note Taking App\\node_modules\\mongoose\\lib\\schematype.js:1202:12)\n    at ObjectId.SchemaType.castForQuery (C:\\Note Taking App\\node_modules\\mongoose\\lib\\schematype.js:1614:15)\n    at cast (C:\\Note Taking App\\node_modules\\mongoose\\lib\\cast.js:348:32)\n    at model.Query.Query.cast (C:\\Note Taking App\\node_modules\\mongoose\\lib\\query.js:4819:12)\n    at model.Query.Query._castConditions (C:\\Note Taking App\\node_modules\\mongoose\\lib\\query.js:2193:10)\n    at model.Query._findOne (C:\\Note Taking App\\node_modules\\mongoose\\lib\\query.js:2477:8)\n    at model.Query.exec (C:\\Note Taking App\\node_modules\\mongoose\\lib\\query.js:4349:28)\n    at processTicksAndRejections (node:internal/process/task_queues:96:5)\n    at async UserManagement.getUserById (C:\\Note Taking App\\src\\services\\userServices.js:11:18)\n    at async getUserById (C:\\Note Taking App\\src\\controllers\\userController.js:51:18)","stringValue":"\"123\"","timestamp":"2023-05-22T17:53:35.543Z","value":"123","valueType":"string"}
{"kind":"ObjectId","level":"error","message":"An error occurred while fetching user by ID: Cast to ObjectId failed for value \"123\" (type string) at path \"_id\" for model \"User\"","path":"_id","reason":{},"stack":"CastError: Cast to ObjectId failed for value \"123\" (type string) at path \"_id\" for model \"User\"\n    at ObjectId.cast (C:\\Note Taking App\\node_modules\\mongoose\\lib\\schema\\objectid.js:248:11)\n    at ObjectId.SchemaType.applySetters (C:\\Note Taking App\\node_modules\\mongoose\\lib\\schematype.js:1202:12)\n    at ObjectId.SchemaType.castForQuery (C:\\Note Taking App\\node_modules\\mongoose\\lib\\schematype.js:1614:15)\n    at cast (C:\\Note Taking App\\node_modules\\mongoose\\lib\\cast.js:348:32)\n    at model.Query.Query.cast (C:\\Note Taking App\\node_modules\\mongoose\\lib\\query.js:4819:12)\n    at model.Query.Query._castConditions (C:\\Note Taking App\\node_modules\\mongoose\\lib\\query.js:2193:10)\n    at model.Query._findOne (C:\\Note Taking App\\node_modules\\mongoose\\lib\\query.js:2477:8)\n    at model.Query.exec (C:\\Note Taking App\\node_modules\\mongoose\\lib\\query.js:4349:28)\n    at processTicksAndRejections (node:internal/process/task_queues:96:5)\n    at async UserManagement.getUserById (C:\\Note Taking App\\src\\services\\userServices.js:11:18)\n    at async getUserById (C:\\Note Taking App\\src\\controllers\\userController.js:51:18)","stringValue":"\"123\"","timestamp":"2023-05-22T17:53:35.574Z","value":"123","valueType":"string"}
{"kind":"ObjectId","level":"error","message":"An error occurred while updating a user: Cast to ObjectId failed for value \"123\" (type string) at path \"_id\" for model \"User\"","path":"_id","reason":{},"stack":"CastError: Cast to ObjectId failed for value \"123\" (type string) at path \"_id\" for model \"User\"\n    at ObjectId.cast (C:\\Note Taking App\\node_modules\\mongoose\\lib\\schema\\objectid.js:248:11)\n    at ObjectId.SchemaType.applySetters (C:\\Note Taking App\\node_modules\\mongoose\\lib\\schematype.js:1202:12)\n    at ObjectId.SchemaType.castForQuery (C:\\Note Taking App\\node_modules\\mongoose\\lib\\schematype.js:1614:15)\n    at cast (C:\\Note Taking App\\node_modules\\mongoose\\lib\\cast.js:348:32)\n    at model.Query.Query.cast (C:\\Note Taking App\\node_modules\\mongoose\\lib\\query.js:4819:12)\n    at model.Query.Query._castConditions (C:\\Note Taking App\\node_modules\\mongoose\\lib\\query.js:2193:10)\n    at model.Query._findOneAndUpdate (C:\\Note Taking App\\node_modules\\mongoose\\lib\\query.js:3223:8)\n    at model.Query.exec (C:\\Note Taking App\\node_modules\\mongoose\\lib\\query.js:4349:28)\n    at processTicksAndRejections (node:internal/process/task_queues:96:5)\n    at async UserManagement.updateUser (C:\\Note Taking App\\src\\services\\userServices.js:17:18)\n    at async updateUser (C:\\Note Taking App\\src\\controllers\\userController.js:86:25)","stringValue":"\"123\"","timestamp":"2023-05-22T17:53:35.610Z","value":"123","valueType":"string"}
{"kind":"ObjectId","level":"error","message":"An error occurred while updating a user: Cast to ObjectId failed for value \"123\" (type string) at path \"_id\" for model \"User\"","path":"_id","reason":{},"stack":"CastError: Cast to ObjectId failed for value \"123\" (type string) at path \"_id\" for model \"User\"\n    at ObjectId.cast (C:\\Note Taking App\\node_modules\\mongoose\\lib\\schema\\objectid.js:248:11)\n    at ObjectId.SchemaType.applySetters (C:\\Note Taking App\\node_modules\\mongoose\\lib\\schematype.js:1202:12)\n    at ObjectId.SchemaType.castForQuery (C:\\Note Taking App\\node_modules\\mongoose\\lib\\schematype.js:1614:15)\n    at cast (C:\\Note Taking App\\node_modules\\mongoose\\lib\\cast.js:348:32)\n    at model.Query.Query.cast (C:\\Note Taking App\\node_modules\\mongoose\\lib\\query.js:4819:12)\n    at model.Query.Query._castConditions (C:\\Note Taking App\\node_modules\\mongoose\\lib\\query.js:2193:10)\n    at model.Query._findOneAndUpdate (C:\\Note Taking App\\node_modules\\mongoose\\lib\\query.js:3223:8)\n    at model.Query.exec (C:\\Note Taking App\\node_modules\\mongoose\\lib\\query.js:4349:28)\n    at processTicksAndRejections (node:internal/process/task_queues:96:5)\n    at async UserManagement.updateUser (C:\\Note Taking App\\src\\services\\userServices.js:17:18)\n    at async updateUser (C:\\Note Taking App\\src\\controllers\\userController.js:86:25)","stringValue":"\"123\"","timestamp":"2023-05-22T17:53:35.646Z","value":"123","valueType":"string"}
{"kind":"ObjectId","level":"error","message":"An error occurred while deleting a user: Cast to ObjectId failed for value \"123\" (type string) at path \"_id\" for model \"User\"","path":"_id","reason":{},"stack":"CastError: Cast to ObjectId failed for value \"123\" (type string) at path \"_id\" for model \"User\"\n    at ObjectId.cast (C:\\Note Taking App\\node_modules\\mongoose\\lib\\schema\\objectid.js:248:11)\n    at ObjectId.SchemaType.applySetters (C:\\Note Taking App\\node_modules\\mongoose\\lib\\schematype.js:1202:12)\n    at ObjectId.SchemaType.castForQuery (C:\\Note Taking App\\node_modules\\mongoose\\lib\\schematype.js:1614:15)\n    at cast (C:\\Note Taking App\\node_modules\\mongoose\\lib\\cast.js:348:32)\n    at model.Query.Query.cast (C:\\Note Taking App\\node_modules\\mongoose\\lib\\query.js:4819:12)\n    at model.Query.Query._castConditions (C:\\Note Taking App\\node_modules\\mongoose\\lib\\query.js:2193:10)\n    at model.Query._findOneAndDelete (C:\\Note Taking App\\node_modules\\mongoose\\lib\\query.js:3422:8)\n    at model.Query._findOneAndRemove (C:\\Note Taking App\\node_modules\\mongoose\\lib\\query.js:3650:15)\n    at model.Query.exec (C:\\Note Taking App\\node_modules\\mongoose\\lib\\query.js:4349:28)\n    at processTicksAndRejections (node:internal/process/task_queues:96:5)\n    at async UserManagement.deleteUser (C:\\Note Taking App\\src\\services\\userServices.js:22:5)\n    at async deleteUser (C:\\Note Taking App\\src\\controllers\\userController.js:106:5)","stringValue":"\"123\"","timestamp":"2023-05-22T17:53:35.673Z","value":"123","valueType":"string"}
{"kind":"ObjectId","level":"error","message":"An error occurred while deleting a user: Cast to ObjectId failed for value \"123\" (type string) at path \"_id\" for model \"User\"","path":"_id","reason":{},"stack":"CastError: Cast to ObjectId failed for value \"123\" (type string) at path \"_id\" for model \"User\"\n    at ObjectId.cast (C:\\Note Taking App\\node_modules\\mongoose\\lib\\schema\\objectid.js:248:11)\n    at ObjectId.SchemaType.applySetters (C:\\Note Taking App\\node_modules\\mongoose\\lib\\schematype.js:1202:12)\n    at ObjectId.SchemaType.castForQuery (C:\\Note Taking App\\node_modules\\mongoose\\lib\\schematype.js:1614:15)\n    at cast (C:\\Note Taking App\\node_modules\\mongoose\\lib\\cast.js:348:32)\n    at model.Query.Query.cast (C:\\Note Taking App\\node_modules\\mongoose\\lib\\query.js:4819:12)\n    at model.Query.Query._castConditions (C:\\Note Taking App\\node_modules\\mongoose\\lib\\query.js:2193:10)\n    at model.Query._findOneAndDelete (C:\\Note Taking App\\node_modules\\mongoose\\lib\\query.js:3422:8)\n    at model.Query._findOneAndRemove (C:\\Note Taking App\\node_modules\\mongoose\\lib\\query.js:3650:15)\n    at model.Query.exec (C:\\Note Taking App\\node_modules\\mongoose\\lib\\query.js:4349:28)\n    at processTicksAndRejections (node:internal/process/task_queues:96:5)\n    at async UserManagement.deleteUser (C:\\Note Taking App\\src\\services\\userServices.js:22:5)\n    at async deleteUser (C:\\Note Taking App\\src\\controllers\\userController.js:106:5)","stringValue":"\"123\"","timestamp":"2023-05-22T17:53:35.695Z","value":"123","valueType":"string"}
{"kind":"ObjectId","level":"error","message":"An error occurred while fetching user by ID: Cast to ObjectId failed for value \"123\" (type string) at path \"_id\" for model \"User\"","path":"_id","reason":{},"stack":"CastError: Cast to ObjectId failed for value \"123\" (type string) at path \"_id\" for model \"User\"\n    at ObjectId.cast (C:\\Note Taking App\\node_modules\\mongoose\\lib\\schema\\objectid.js:248:11)\n    at ObjectId.SchemaType.applySetters (C:\\Note Taking App\\node_modules\\mongoose\\lib\\schematype.js:1202:12)\n    at ObjectId.SchemaType.castForQuery (C:\\Note Taking App\\node_modules\\mongoose\\lib\\schematype.js:1614:15)\n    at cast (C:\\Note Taking App\\node_modules\\mongoose\\lib\\cast.js:348:32)\n    at model.Query.Query.cast (C:\\Note Taking App\\node_modules\\mongoose\\lib\\query.js:4819:12)\n    at model.Query.Query._castConditions (C:\\Note Taking App\\node_modules\\mongoose\\lib\\query.js:2193:10)\n    at model.Query._findOne (C:\\Note Taking App\\node_modules\\mongoose\\lib\\query.js:2477:8)\n    at model.Query.exec (C:\\Note Taking App\\node_modules\\mongoose\\lib\\query.js:4349:28)\n    at processTicksAndRejections (node:internal/process/task_queues:96:5)\n    at async UserManagement.getUserById (C:\\Note Taking App\\src\\services\\userServices.js:11:18)\n    at async getUserById (C:\\Note Taking App\\src\\controllers\\userController.js:51:18)","stringValue":"\"123\"","timestamp":"2023-05-22T18:00:46.853Z","value":"123","valueType":"string"}
{"kind":"ObjectId","level":"error","message":"An error occurred while fetching user by ID: Cast to ObjectId failed for value \"123\" (type string) at path \"_id\" for model \"User\"","path":"_id","reason":{},"stack":"CastError: Cast to ObjectId failed for value \"123\" (type string) at path \"_id\" for model \"User\"\n    at ObjectId.cast (C:\\Note Taking App\\node_modules\\mongoose\\lib\\schema\\objectid.js:248:11)\n    at ObjectId.SchemaType.applySetters (C:\\Note Taking App\\node_modules\\mongoose\\lib\\schematype.js:1202:12)\n    at ObjectId.SchemaType.castForQuery (C:\\Note Taking App\\node_modules\\mongoose\\lib\\schematype.js:1614:15)\n    at cast (C:\\Note Taking App\\node_modules\\mongoose\\lib\\cast.js:348:32)\n    at model.Query.Query.cast (C:\\Note Taking App\\node_modules\\mongoose\\lib\\query.js:4819:12)\n    at model.Query.Query._castConditions (C:\\Note Taking App\\node_modules\\mongoose\\lib\\query.js:2193:10)\n    at model.Query._findOne (C:\\Note Taking App\\node_modules\\mongoose\\lib\\query.js:2477:8)\n    at model.Query.exec (C:\\Note Taking App\\node_modules\\mongoose\\lib\\query.js:4349:28)\n    at processTicksAndRejections (node:internal/process/task_queues:96:5)\n    at async UserManagement.getUserById (C:\\Note Taking App\\src\\services\\userServices.js:11:18)\n    at async getUserById (C:\\Note Taking App\\src\\controllers\\userController.js:51:18)","stringValue":"\"123\"","timestamp":"2023-05-22T18:00:46.883Z","value":"123","valueType":"string"}
{"kind":"ObjectId","level":"error","message":"An error occurred while updating a user: Cast to ObjectId failed for value \"123\" (type string) at path \"_id\" for model \"User\"","path":"_id","reason":{},"stack":"CastError: Cast to ObjectId failed for value \"123\" (type string) at path \"_id\" for model \"User\"\n    at ObjectId.cast (C:\\Note Taking App\\node_modules\\mongoose\\lib\\schema\\objectid.js:248:11)\n    at ObjectId.SchemaType.applySetters (C:\\Note Taking App\\node_modules\\mongoose\\lib\\schematype.js:1202:12)\n    at ObjectId.SchemaType.castForQuery (C:\\Note Taking App\\node_modules\\mongoose\\lib\\schematype.js:1614:15)\n    at cast (C:\\Note Taking App\\node_modules\\mongoose\\lib\\cast.js:348:32)\n    at model.Query.Query.cast (C:\\Note Taking App\\node_modules\\mongoose\\lib\\query.js:4819:12)\n    at model.Query.Query._castConditions (C:\\Note Taking App\\node_modules\\mongoose\\lib\\query.js:2193:10)\n    at model.Query._findOneAndUpdate (C:\\Note Taking App\\node_modules\\mongoose\\lib\\query.js:3223:8)\n    at model.Query.exec (C:\\Note Taking App\\node_modules\\mongoose\\lib\\query.js:4349:28)\n    at processTicksAndRejections (node:internal/process/task_queues:96:5)\n    at async UserManagement.updateUser (C:\\Note Taking App\\src\\services\\userServices.js:17:18)\n    at async updateUser (C:\\Note Taking App\\src\\controllers\\userController.js:86:25)","stringValue":"\"123\"","timestamp":"2023-05-22T18:00:46.917Z","value":"123","valueType":"string"}
{"kind":"ObjectId","level":"error","message":"An error occurred while updating a user: Cast to ObjectId failed for value \"123\" (type string) at path \"_id\" for model \"User\"","path":"_id","reason":{},"stack":"CastError: Cast to ObjectId failed for value \"123\" (type string) at path \"_id\" for model \"User\"\n    at ObjectId.cast (C:\\Note Taking App\\node_modules\\mongoose\\lib\\schema\\objectid.js:248:11)\n    at ObjectId.SchemaType.applySetters (C:\\Note Taking App\\node_modules\\mongoose\\lib\\schematype.js:1202:12)\n    at ObjectId.SchemaType.castForQuery (C:\\Note Taking App\\node_modules\\mongoose\\lib\\schematype.js:1614:15)\n    at cast (C:\\Note Taking App\\node_modules\\mongoose\\lib\\cast.js:348:32)\n    at model.Query.Query.cast (C:\\Note Taking App\\node_modules\\mongoose\\lib\\query.js:4819:12)\n    at model.Query.Query._castConditions (C:\\Note Taking App\\node_modules\\mongoose\\lib\\query.js:2193:10)\n    at model.Query._findOneAndUpdate (C:\\Note Taking App\\node_modules\\mongoose\\lib\\query.js:3223:8)\n    at model.Query.exec (C:\\Note Taking App\\node_modules\\mongoose\\lib\\query.js:4349:28)\n    at processTicksAndRejections (node:internal/process/task_queues:96:5)\n    at async UserManagement.updateUser (C:\\Note Taking App\\src\\services\\userServices.js:17:18)\n    at async updateUser (C:\\Note Taking App\\src\\controllers\\userController.js:86:25)","stringValue":"\"123\"","timestamp":"2023-05-22T18:00:46.942Z","value":"123","valueType":"string"}
{"kind":"ObjectId","level":"error","message":"An error occurred while deleting a user: Cast to ObjectId failed for value \"123\" (type string) at path \"_id\" for model \"User\"","path":"_id","reason":{},"stack":"CastError: Cast to ObjectId failed for value \"123\" (type string) at path \"_id\" for model \"User\"\n    at ObjectId.cast (C:\\Note Taking App\\node_modules\\mongoose\\lib\\schema\\objectid.js:248:11)\n    at ObjectId.SchemaType.applySetters (C:\\Note Taking App\\node_modules\\mongoose\\lib\\schematype.js:1202:12)\n    at ObjectId.SchemaType.castForQuery (C:\\Note Taking App\\node_modules\\mongoose\\lib\\schematype.js:1614:15)\n    at cast (C:\\Note Taking App\\node_modules\\mongoose\\lib\\cast.js:348:32)\n    at model.Query.Query.cast (C:\\Note Taking App\\node_modules\\mongoose\\lib\\query.js:4819:12)\n    at model.Query.Query._castConditions (C:\\Note Taking App\\node_modules\\mongoose\\lib\\query.js:2193:10)\n    at model.Query._findOneAndDelete (C:\\Note Taking App\\node_modules\\mongoose\\lib\\query.js:3422:8)\n    at model.Query._findOneAndRemove (C:\\Note Taking App\\node_modules\\mongoose\\lib\\query.js:3650:15)\n    at model.Query.exec (C:\\Note Taking App\\node_modules\\mongoose\\lib\\query.js:4349:28)\n    at processTicksAndRejections (node:internal/process/task_queues:96:5)\n    at async UserManagement.deleteUser (C:\\Note Taking App\\src\\services\\userServices.js:22:5)\n    at async deleteUser (C:\\Note Taking App\\src\\controllers\\userController.js:106:5)","stringValue":"\"123\"","timestamp":"2023-05-22T18:00:46.967Z","value":"123","valueType":"string"}
{"kind":"ObjectId","level":"error","message":"An error occurred while deleting a user: Cast to ObjectId failed for value \"123\" (type string) at path \"_id\" for model \"User\"","path":"_id","reason":{},"stack":"CastError: Cast to ObjectId failed for value \"123\" (type string) at path \"_id\" for model \"User\"\n    at ObjectId.cast (C:\\Note Taking App\\node_modules\\mongoose\\lib\\schema\\objectid.js:248:11)\n    at ObjectId.SchemaType.applySetters (C:\\Note Taking App\\node_modules\\mongoose\\lib\\schematype.js:1202:12)\n    at ObjectId.SchemaType.castForQuery (C:\\Note Taking App\\node_modules\\mongoose\\lib\\schematype.js:1614:15)\n    at cast (C:\\Note Taking App\\node_modules\\mongoose\\lib\\cast.js:348:32)\n    at model.Query.Query.cast (C:\\Note Taking App\\node_modules\\mongoose\\lib\\query.js:4819:12)\n    at model.Query.Query._castConditions (C:\\Note Taking App\\node_modules\\mongoose\\lib\\query.js:2193:10)\n    at model.Query._findOneAndDelete (C:\\Note Taking App\\node_modules\\mongoose\\lib\\query.js:3422:8)\n    at model.Query._findOneAndRemove (C:\\Note Taking App\\node_modules\\mongoose\\lib\\query.js:3650:15)\n    at model.Query.exec (C:\\Note Taking App\\node_modules\\mongoose\\lib\\query.js:4349:28)\n    at processTicksAndRejections (node:internal/process/task_queues:96:5)\n    at async UserManagement.deleteUser (C:\\Note Taking App\\src\\services\\userServices.js:22:5)\n    at async deleteUser (C:\\Note Taking App\\src\\controllers\\userController.js:106:5)","stringValue":"\"123\"","timestamp":"2023-05-22T18:00:47.000Z","value":"123","valueType":"string"}
{"level":"error","message":"An error occurred while creating a user: res.status(...).message is not a function","stack":"TypeError: res.status(...).message is not a function\n    at createUser (C:\\Note Taking App\\src\\controllers\\userController.js:18:24)\n    at processTicksAndRejections (node:internal/process/task_queues:96:5)","timestamp":"2023-05-22T18:04:41.397Z"}
{"kind":"ObjectId","level":"error","message":"An error occurred while fetching user by ID: Cast to ObjectId failed for value \"123\" (type string) at path \"_id\" for model \"User\"","path":"_id","reason":{},"stack":"CastError: Cast to ObjectId failed for value \"123\" (type string) at path \"_id\" for model \"User\"\n    at ObjectId.cast (C:\\Note Taking App\\node_modules\\mongoose\\lib\\schema\\objectid.js:248:11)\n    at ObjectId.SchemaType.applySetters (C:\\Note Taking App\\node_modules\\mongoose\\lib\\schematype.js:1202:12)\n    at ObjectId.SchemaType.castForQuery (C:\\Note Taking App\\node_modules\\mongoose\\lib\\schematype.js:1614:15)\n    at cast (C:\\Note Taking App\\node_modules\\mongoose\\lib\\cast.js:348:32)\n    at model.Query.Query.cast (C:\\Note Taking App\\node_modules\\mongoose\\lib\\query.js:4819:12)\n    at model.Query.Query._castConditions (C:\\Note Taking App\\node_modules\\mongoose\\lib\\query.js:2193:10)\n    at model.Query._findOne (C:\\Note Taking App\\node_modules\\mongoose\\lib\\query.js:2477:8)\n    at model.Query.exec (C:\\Note Taking App\\node_modules\\mongoose\\lib\\query.js:4349:28)\n    at processTicksAndRejections (node:internal/process/task_queues:96:5)\n    at async UserManagement.getUserById (C:\\Note Taking App\\src\\services\\userServices.js:11:18)\n    at async getUserById (C:\\Note Taking App\\src\\controllers\\userController.js:51:18)","stringValue":"\"123\"","timestamp":"2023-05-22T18:04:41.426Z","value":"123","valueType":"string"}
{"kind":"ObjectId","level":"error","message":"An error occurred while fetching user by ID: Cast to ObjectId failed for value \"123\" (type string) at path \"_id\" for model \"User\"","path":"_id","reason":{},"stack":"CastError: Cast to ObjectId failed for value \"123\" (type string) at path \"_id\" for model \"User\"\n    at ObjectId.cast (C:\\Note Taking App\\node_modules\\mongoose\\lib\\schema\\objectid.js:248:11)\n    at ObjectId.SchemaType.applySetters (C:\\Note Taking App\\node_modules\\mongoose\\lib\\schematype.js:1202:12)\n    at ObjectId.SchemaType.castForQuery (C:\\Note Taking App\\node_modules\\mongoose\\lib\\schematype.js:1614:15)\n    at cast (C:\\Note Taking App\\node_modules\\mongoose\\lib\\cast.js:348:32)\n    at model.Query.Query.cast (C:\\Note Taking App\\node_modules\\mongoose\\lib\\query.js:4819:12)\n    at model.Query.Query._castConditions (C:\\Note Taking App\\node_modules\\mongoose\\lib\\query.js:2193:10)\n    at model.Query._findOne (C:\\Note Taking App\\node_modules\\mongoose\\lib\\query.js:2477:8)\n    at model.Query.exec (C:\\Note Taking App\\node_modules\\mongoose\\lib\\query.js:4349:28)\n    at processTicksAndRejections (node:internal/process/task_queues:96:5)\n    at async UserManagement.getUserById (C:\\Note Taking App\\src\\services\\userServices.js:11:18)\n    at async getUserById (C:\\Note Taking App\\src\\controllers\\userController.js:51:18)","stringValue":"\"123\"","timestamp":"2023-05-22T18:04:41.452Z","value":"123","valueType":"string"}
{"kind":"ObjectId","level":"error","message":"An error occurred while updating a user: Cast to ObjectId failed for value \"123\" (type string) at path \"_id\" for model \"User\"","path":"_id","reason":{},"stack":"CastError: Cast to ObjectId failed for value \"123\" (type string) at path \"_id\" for model \"User\"\n    at ObjectId.cast (C:\\Note Taking App\\node_modules\\mongoose\\lib\\schema\\objectid.js:248:11)\n    at ObjectId.SchemaType.applySetters (C:\\Note Taking App\\node_modules\\mongoose\\lib\\schematype.js:1202:12)\n    at ObjectId.SchemaType.castForQuery (C:\\Note Taking App\\node_modules\\mongoose\\lib\\schematype.js:1614:15)\n    at cast (C:\\Note Taking App\\node_modules\\mongoose\\lib\\cast.js:348:32)\n    at model.Query.Query.cast (C:\\Note Taking App\\node_modules\\mongoose\\lib\\query.js:4819:12)\n    at model.Query.Query._castConditions (C:\\Note Taking App\\node_modules\\mongoose\\lib\\query.js:2193:10)\n    at model.Query._findOneAndUpdate (C:\\Note Taking App\\node_modules\\mongoose\\lib\\query.js:3223:8)\n    at model.Query.exec (C:\\Note Taking App\\node_modules\\mongoose\\lib\\query.js:4349:28)\n    at processTicksAndRejections (node:internal/process/task_queues:96:5)\n    at async UserManagement.updateUser (C:\\Note Taking App\\src\\services\\userServices.js:17:18)\n    at async updateUser (C:\\Note Taking App\\src\\controllers\\userController.js:86:25)","stringValue":"\"123\"","timestamp":"2023-05-22T18:04:41.492Z","value":"123","valueType":"string"}
{"kind":"ObjectId","level":"error","message":"An error occurred while updating a user: Cast to ObjectId failed for value \"123\" (type string) at path \"_id\" for model \"User\"","path":"_id","reason":{},"stack":"CastError: Cast to ObjectId failed for value \"123\" (type string) at path \"_id\" for model \"User\"\n    at ObjectId.cast (C:\\Note Taking App\\node_modules\\mongoose\\lib\\schema\\objectid.js:248:11)\n    at ObjectId.SchemaType.applySetters (C:\\Note Taking App\\node_modules\\mongoose\\lib\\schematype.js:1202:12)\n    at ObjectId.SchemaType.castForQuery (C:\\Note Taking App\\node_modules\\mongoose\\lib\\schematype.js:1614:15)\n    at cast (C:\\Note Taking App\\node_modules\\mongoose\\lib\\cast.js:348:32)\n    at model.Query.Query.cast (C:\\Note Taking App\\node_modules\\mongoose\\lib\\query.js:4819:12)\n    at model.Query.Query._castConditions (C:\\Note Taking App\\node_modules\\mongoose\\lib\\query.js:2193:10)\n    at model.Query._findOneAndUpdate (C:\\Note Taking App\\node_modules\\mongoose\\lib\\query.js:3223:8)\n    at model.Query.exec (C:\\Note Taking App\\node_modules\\mongoose\\lib\\query.js:4349:28)\n    at processTicksAndRejections (node:internal/process/task_queues:96:5)\n    at async UserManagement.updateUser (C:\\Note Taking App\\src\\services\\userServices.js:17:18)\n    at async updateUser (C:\\Note Taking App\\src\\controllers\\userController.js:86:25)","stringValue":"\"123\"","timestamp":"2023-05-22T18:04:41.523Z","value":"123","valueType":"string"}
{"kind":"ObjectId","level":"error","message":"An error occurred while deleting a user: Cast to ObjectId failed for value \"123\" (type string) at path \"_id\" for model \"User\"","path":"_id","reason":{},"stack":"CastError: Cast to ObjectId failed for value \"123\" (type string) at path \"_id\" for model \"User\"\n    at ObjectId.cast (C:\\Note Taking App\\node_modules\\mongoose\\lib\\schema\\objectid.js:248:11)\n    at ObjectId.SchemaType.applySetters (C:\\Note Taking App\\node_modules\\mongoose\\lib\\schematype.js:1202:12)\n    at ObjectId.SchemaType.castForQuery (C:\\Note Taking App\\node_modules\\mongoose\\lib\\schematype.js:1614:15)\n    at cast (C:\\Note Taking App\\node_modules\\mongoose\\lib\\cast.js:348:32)\n    at model.Query.Query.cast (C:\\Note Taking App\\node_modules\\mongoose\\lib\\query.js:4819:12)\n    at model.Query.Query._castConditions (C:\\Note Taking App\\node_modules\\mongoose\\lib\\query.js:2193:10)\n    at model.Query._findOneAndDelete (C:\\Note Taking App\\node_modules\\mongoose\\lib\\query.js:3422:8)\n    at model.Query._findOneAndRemove (C:\\Note Taking App\\node_modules\\mongoose\\lib\\query.js:3650:15)\n    at model.Query.exec (C:\\Note Taking App\\node_modules\\mongoose\\lib\\query.js:4349:28)\n    at processTicksAndRejections (node:internal/process/task_queues:96:5)\n    at async UserManagement.deleteUser (C:\\Note Taking App\\src\\services\\userServices.js:22:5)\n    at async deleteUser (C:\\Note Taking App\\src\\controllers\\userController.js:106:5)","stringValue":"\"123\"","timestamp":"2023-05-22T18:04:41.550Z","value":"123","valueType":"string"}
{"kind":"ObjectId","level":"error","message":"An error occurred while deleting a user: Cast to ObjectId failed for value \"123\" (type string) at path \"_id\" for model \"User\"","path":"_id","reason":{},"stack":"CastError: Cast to ObjectId failed for value \"123\" (type string) at path \"_id\" for model \"User\"\n    at ObjectId.cast (C:\\Note Taking App\\node_modules\\mongoose\\lib\\schema\\objectid.js:248:11)\n    at ObjectId.SchemaType.applySetters (C:\\Note Taking App\\node_modules\\mongoose\\lib\\schematype.js:1202:12)\n    at ObjectId.SchemaType.castForQuery (C:\\Note Taking App\\node_modules\\mongoose\\lib\\schematype.js:1614:15)\n    at cast (C:\\Note Taking App\\node_modules\\mongoose\\lib\\cast.js:348:32)\n    at model.Query.Query.cast (C:\\Note Taking App\\node_modules\\mongoose\\lib\\query.js:4819:12)\n    at model.Query.Query._castConditions (C:\\Note Taking App\\node_modules\\mongoose\\lib\\query.js:2193:10)\n    at model.Query._findOneAndDelete (C:\\Note Taking App\\node_modules\\mongoose\\lib\\query.js:3422:8)\n    at model.Query._findOneAndRemove (C:\\Note Taking App\\node_modules\\mongoose\\lib\\query.js:3650:15)\n    at model.Query.exec (C:\\Note Taking App\\node_modules\\mongoose\\lib\\query.js:4349:28)\n    at processTicksAndRejections (node:internal/process/task_queues:96:5)\n    at async UserManagement.deleteUser (C:\\Note Taking App\\src\\services\\userServices.js:22:5)\n    at async deleteUser (C:\\Note Taking App\\src\\controllers\\userController.js:106:5)","stringValue":"\"123\"","timestamp":"2023-05-22T18:04:41.573Z","value":"123","valueType":"string"}
{"level":"error","message":"An error occurred while creating a user: res.status(...).message is not a function","stack":"TypeError: res.status(...).message is not a function\n    at createUser (C:\\Note Taking App\\src\\controllers\\userController.js:19:24)\n    at processTicksAndRejections (node:internal/process/task_queues:96:5)","timestamp":"2023-05-22T18:30:46.988Z"}
{"kind":"ObjectId","level":"error","message":"An error occurred while fetching user by ID: Cast to ObjectId failed for value \"undefined\" (type string) at path \"_id\" for model \"User\"","path":"_id","reason":{},"stack":"CastError: Cast to ObjectId failed for value \"undefined\" (type string) at path \"_id\" for model \"User\"\n    at ObjectId.cast (C:\\Note Taking App\\node_modules\\mongoose\\lib\\schema\\objectid.js:248:11)\n    at ObjectId.SchemaType.applySetters (C:\\Note Taking App\\node_modules\\mongoose\\lib\\schematype.js:1202:12)\n    at ObjectId.SchemaType.castForQuery (C:\\Note Taking App\\node_modules\\mongoose\\lib\\schematype.js:1614:15)\n    at cast (C:\\Note Taking App\\node_modules\\mongoose\\lib\\cast.js:348:32)\n    at model.Query.Query.cast (C:\\Note Taking App\\node_modules\\mongoose\\lib\\query.js:4819:12)\n    at model.Query.Query._castConditions (C:\\Note Taking App\\node_modules\\mongoose\\lib\\query.js:2193:10)\n    at model.Query._findOne (C:\\Note Taking App\\node_modules\\mongoose\\lib\\query.js:2477:8)\n    at model.Query.exec (C:\\Note Taking App\\node_modules\\mongoose\\lib\\query.js:4349:28)\n    at processTicksAndRejections (node:internal/process/task_queues:96:5)\n    at async UserManagement.getUserById (C:\\Note Taking App\\src\\services\\userServices.js:11:18)\n    at async getUserById (C:\\Note Taking App\\src\\controllers\\userController.js:53:18)","stringValue":"\"undefined\"","timestamp":"2023-05-22T18:30:47.025Z","value":"undefined","valueType":"string"}
{"kind":"ObjectId","level":"error","message":"An error occurred while updating a user: Cast to ObjectId failed for value \"undefined\" (type string) at path \"_id\" for model \"User\"","path":"_id","reason":{},"stack":"CastError: Cast to ObjectId failed for value \"undefined\" (type string) at path \"_id\" for model \"User\"\n    at ObjectId.cast (C:\\Note Taking App\\node_modules\\mongoose\\lib\\schema\\objectid.js:248:11)\n    at ObjectId.SchemaType.applySetters (C:\\Note Taking App\\node_modules\\mongoose\\lib\\schematype.js:1202:12)\n    at ObjectId.SchemaType.castForQuery (C:\\Note Taking App\\node_modules\\mongoose\\lib\\schematype.js:1614:15)\n    at cast (C:\\Note Taking App\\node_modules\\mongoose\\lib\\cast.js:348:32)\n    at model.Query.Query.cast (C:\\Note Taking App\\node_modules\\mongoose\\lib\\query.js:4819:12)\n    at model.Query.Query._castConditions (C:\\Note Taking App\\node_modules\\mongoose\\lib\\query.js:2193:10)\n    at model.Query._findOneAndUpdate (C:\\Note Taking App\\node_modules\\mongoose\\lib\\query.js:3223:8)\n    at model.Query.exec (C:\\Note Taking App\\node_modules\\mongoose\\lib\\query.js:4349:28)\n    at processTicksAndRejections (node:internal/process/task_queues:96:5)\n    at async UserManagement.updateUser (C:\\Note Taking App\\src\\services\\userServices.js:17:18)\n    at async updateUser (C:\\Note Taking App\\src\\controllers\\userController.js:88:25)","stringValue":"\"undefined\"","timestamp":"2023-05-22T18:30:47.115Z","value":"undefined","valueType":"string"}
{"kind":"ObjectId","level":"error","message":"An error occurred while deleting a user: Cast to ObjectId failed for value \"undefined\" (type string) at path \"_id\" for model \"User\"","path":"_id","reason":{},"stack":"CastError: Cast to ObjectId failed for value \"undefined\" (type string) at path \"_id\" for model \"User\"\n    at ObjectId.cast (C:\\Note Taking App\\node_modules\\mongoose\\lib\\schema\\objectid.js:248:11)\n    at ObjectId.SchemaType.applySetters (C:\\Note Taking App\\node_modules\\mongoose\\lib\\schematype.js:1202:12)\n    at ObjectId.SchemaType.castForQuery (C:\\Note Taking App\\node_modules\\mongoose\\lib\\schematype.js:1614:15)\n    at cast (C:\\Note Taking App\\node_modules\\mongoose\\lib\\cast.js:348:32)\n    at model.Query.Query.cast (C:\\Note Taking App\\node_modules\\mongoose\\lib\\query.js:4819:12)\n    at model.Query.Query._castConditions (C:\\Note Taking App\\node_modules\\mongoose\\lib\\query.js:2193:10)\n    at model.Query._findOneAndDelete (C:\\Note Taking App\\node_modules\\mongoose\\lib\\query.js:3422:8)\n    at model.Query._findOneAndRemove (C:\\Note Taking App\\node_modules\\mongoose\\lib\\query.js:3650:15)\n    at model.Query.exec (C:\\Note Taking App\\node_modules\\mongoose\\lib\\query.js:4349:28)\n    at processTicksAndRejections (node:internal/process/task_queues:96:5)\n    at async UserManagement.deleteUser (C:\\Note Taking App\\src\\services\\userServices.js:22:5)\n    at async deleteUser (C:\\Note Taking App\\src\\controllers\\userController.js:108:5)","stringValue":"\"undefined\"","timestamp":"2023-05-22T18:30:47.200Z","value":"undefined","valueType":"string"}
{"kind":"ObjectId","level":"error","message":"An error occurred while deleting a user: Cast to ObjectId failed for value \"123\" (type string) at path \"_id\" for model \"User\"","path":"_id","reason":{},"stack":"CastError: Cast to ObjectId failed for value \"123\" (type string) at path \"_id\" for model \"User\"\n    at ObjectId.cast (C:\\Note Taking App\\node_modules\\mongoose\\lib\\schema\\objectid.js:248:11)\n    at ObjectId.SchemaType.applySetters (C:\\Note Taking App\\node_modules\\mongoose\\lib\\schematype.js:1202:12)\n    at ObjectId.SchemaType.castForQuery (C:\\Note Taking App\\node_modules\\mongoose\\lib\\schematype.js:1614:15)\n    at cast (C:\\Note Taking App\\node_modules\\mongoose\\lib\\cast.js:348:32)\n    at model.Query.Query.cast (C:\\Note Taking App\\node_modules\\mongoose\\lib\\query.js:4819:12)\n    at model.Query.Query._castConditions (C:\\Note Taking App\\node_modules\\mongoose\\lib\\query.js:2193:10)\n    at model.Query._findOneAndDelete (C:\\Note Taking App\\node_modules\\mongoose\\lib\\query.js:3422:8)\n    at model.Query._findOneAndRemove (C:\\Note Taking App\\node_modules\\mongoose\\lib\\query.js:3650:15)\n    at model.Query.exec (C:\\Note Taking App\\node_modules\\mongoose\\lib\\query.js:4349:28)\n    at processTicksAndRejections (node:internal/process/task_queues:96:5)\n    at async UserManagement.deleteUser (C:\\Note Taking App\\src\\services\\userServices.js:22:5)\n    at async deleteUser (C:\\Note Taking App\\src\\controllers\\userController.js:108:5)","stringValue":"\"123\"","timestamp":"2023-05-22T19:30:43.939Z","value":"123","valueType":"string"}
{"kind":"ObjectId","level":"error","message":"An error occurred while deleting a user: Cast to ObjectId failed for value \"123\" (type string) at path \"_id\" for model \"User\"","path":"_id","reason":{},"stack":"CastError: Cast to ObjectId failed for value \"123\" (type string) at path \"_id\" for model \"User\"\n    at ObjectId.cast (C:\\Note Taking App\\node_modules\\mongoose\\lib\\schema\\objectid.js:248:11)\n    at ObjectId.SchemaType.applySetters (C:\\Note Taking App\\node_modules\\mongoose\\lib\\schematype.js:1202:12)\n    at ObjectId.SchemaType.castForQuery (C:\\Note Taking App\\node_modules\\mongoose\\lib\\schematype.js:1614:15)\n    at cast (C:\\Note Taking App\\node_modules\\mongoose\\lib\\cast.js:348:32)\n    at model.Query.Query.cast (C:\\Note Taking App\\node_modules\\mongoose\\lib\\query.js:4819:12)\n    at model.Query.Query._castConditions (C:\\Note Taking App\\node_modules\\mongoose\\lib\\query.js:2193:10)\n    at model.Query._findOneAndDelete (C:\\Note Taking App\\node_modules\\mongoose\\lib\\query.js:3422:8)\n    at model.Query._findOneAndRemove (C:\\Note Taking App\\node_modules\\mongoose\\lib\\query.js:3650:15)\n    at model.Query.exec (C:\\Note Taking App\\node_modules\\mongoose\\lib\\query.js:4349:28)\n    at processTicksAndRejections (node:internal/process/task_queues:96:5)\n    at async UserManagement.deleteUser (C:\\Note Taking App\\src\\services\\userServices.js:22:5)\n    at async deleteUser (C:\\Note Taking App\\src\\controllers\\userController.js:108:5)","stringValue":"\"123\"","timestamp":"2023-05-22T19:34:02.322Z","value":"123","valueType":"string"}
